{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_tcell_fine.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra_hg38.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "ylim = 300000\n",
    "line_style = solid\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "#[CTCF NN variants]\n",
    "#file = \n",
    "#title = Variants CTCF perturbing\n",
    "#height = 1\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#labels = false\n",
    "#file_type = bed\n",
    "\n",
    "[ENCFF518JVS_cd8_tcell_imputed]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF518JVS_cd8_tcell_imputed_hg38.bw\n",
    "title = CTCF chip-seq CD8 Tcell imputed\n",
    "height = 3\n",
    "color = #42f44e\n",
    "#overlay previous = yes\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "max_value = 10\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = True\n",
    "file_type = bigwig\n",
    "\n",
    "[CD8_effmemT_CPG_meth]\n",
    "file = /input_dir/corona_analysis/tracks/S002ND51_CD8_effmemT_CPG_meth_hg38.bw\n",
    "title = CD8+ eff memory T-cell C-meth\n",
    "height = .5\n",
    "color = #21ffff\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[T-cell CD8 h3k4me3]\n",
    "file = /input_dir/corona_analysis/tracks/C0054X_cd8_tcell_h3k4me3_hg38.bw\n",
    "title = CD8 T-cell H3k4Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #ffb42b\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-cell CD8 H3k27Ac]\n",
    "file = /input_dir/corona_analysis/tracks/C0054X_cd8_tcell_h3k27Ac_hg38.bw\n",
    "title = CD8 T-cell H3k27Ac fc Chip-seq\n",
    "height = 2\n",
    "color = #42f4f4\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-cell CD8 H3k9Me3]\n",
    "file = /input_dir/corona_analysis/tracks/C0054X_cd8_tcell_h3k9me3_hg38.bw\n",
    "title = CD8 T-cell H3k9Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #a22eea\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell 1 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T1_scATAC_hg38.bw\n",
    "title = [T-Cell 1 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell 2 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T2_scATAC_hg38.bw\n",
    "title = [T-Cell 2 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig       \n",
    "\n",
    "[T-Cell 3 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T3_scATAC_hg38.bw\n",
    "title = T-Cell 3 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell EM cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_EM_T_scATAC_hg38.bw\n",
    "title = T-Cell EM cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell CM cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_CM_T_scATAC_hg38.bw\n",
    "title = T-Cell CM cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         \n",
    "[CD8_MEMORY_PRIMARY_CELLS]\n",
    "#file = /input_dir/corona_analysis/tracks/CD8_MEMORY_PRIMARY_CELLS_hg38.bed.gz\n",
    "file = /input_dir/corona_analysis/tracks/E048_15_coreMarks_dense_hg38.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD8 Tcell memory cells\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[CD8_NAIVE_PRIMARY_CELLS]\n",
    "#file = /input_dir/corona_analysis/tracks/CD8_NAIVE_PRIMARY_CELLS_hg38.bed.gz\n",
    "file = /input_dir/corona_analysis/tracks/E047_15_coreMarks_dense_hg38.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD8 Tcell naive cells\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[COPD GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg38.bed\n",
    "height = 1\n",
    "title = COPD GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed         \n",
    "         \n",
    "[IPF GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed\n",
    "height = 1\n",
    "title = IPF GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed      \n",
    "         \n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "#[ClinVar kidney phenotype]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/clinvar_kidney_variants_hg38.bed\n",
    "#height = 1\n",
    "#title = ClinVar w/ immune phenotype\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#file_type = bed\n",
    "#fontsize = 10\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "[CD8_T_mem_rnaseq_plus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_plus.bw\n",
    "title = RNA CD8 TCell +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         \n",
    "[CD8_T_mem_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_minus.bw\n",
    "title = RNA CD8 TCell -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_control_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF control +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_control_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF control -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_infected_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF CoVid +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_infected_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF CoVid -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "#overlay_previous = share-y\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_tcell_broad.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[CD8 T-Cell1]\n",
    "file=/input_dir/corona_analysis/tracks/GSM2827788_CD8T1_hg_t.cool\n",
    "title = Tcell S1\n",
    "colormap = coolwarm\n",
    "min_value = .2\n",
    "max_value = 5.2\n",
    "depth = 3000000\n",
    "transform = log\n",
    "show_masked_bins = False\n",
    "file_type = hic_matrix\n",
    "height = 10\n",
    "\n",
    "[CD8 T-Cell2]\n",
    "file=/input_dir/corona_analysis/tracks/GSM2827789_CD8T2_hg_t.cool\n",
    "title = Tcell S2\n",
    "colormap = coolwarm\n",
    "min_value = .2\n",
    "max_value = 5.2\n",
    "depth = 3000000\n",
    "transform = log\n",
    "show_masked_bins = False\n",
    "file_type = hic_matrix\n",
    "height = 10\n",
    "\n",
    "#[TAD Calls]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/HEK293T_100kbTADs_domains.bed\n",
    "#file_type = domains\n",
    "#overlay_previous = share-y\n",
    "#fontsize = 8\n",
    "#line_width = 3\n",
    "#border_color = black\n",
    "#color = none\n",
    "\n",
    "[ENCFF518JVS_cd8_tcell_imputed]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF518JVS_cd8_tcell_imputed_hg38.bw\n",
    "title = CTCF chip-seq CD8 Tcell imputed\n",
    "height = 3\n",
    "color = #42f44e\n",
    "#overlay previous = yes\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "max_value = 10\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = True\n",
    "file_type = bigwig\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra_hg38.arcs\n",
    "title = Cohesin Chia-PET in hESC\n",
    "compact_level = 1\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "line_style = solid\n",
    "orientation=inverted\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "[T-Cell 1 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T1_scATAC_hg38.bw\n",
    "title = [T-Cell 1 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell EM cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_EM_T_scATAC_hg38.bw\n",
    "title = T-Cell EM cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[CD8_MEMORY_PRIMARY_CELLS]\n",
    "#file = /input_dir/corona_analysis/tracks/CD8_MEMORY_PRIMARY_CELLS_hg38.bed.gz\n",
    "file = /input_dir/corona_analysis/tracks/E048_15_coreMarks_dense_hg38.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD8 Tcell memory cells\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[COPD GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg38.bed\n",
    "height = 1\n",
    "title = COPD GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed      \n",
    "         \n",
    "[IPF GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed\n",
    "height = 1\n",
    "title = IPF GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "         \n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "         \n",
    "[CD8_T_mem_rnaseq_plus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_plus.bw\n",
    "title = RNA CD8 TCell +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         \n",
    "[CD8_T_mem_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_minus.bw\n",
    "title = RNA CD8 TCell -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IL2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr4:122435776-122467762 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL2_cd8t_hg38_fine.pdf   \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Innate immune"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# T-cell differentiation\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr14:98785652-98836861 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/BCL11b_cd8t_hg38_z1.pdf   \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr14:98669194-99282480 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/BCL11b_cd8t_hg38_z2.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:pygenometracks.tracksClass:In section 5. [H1 Primed CTCF loops], these parameters are unused:['compact_level']\n",
      "INFO:pygenometracks.tracksClass:initialize 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:initialize 2. [CD8 T-Cell1]\n",
      "WARNING:hicmatrix.HiCMatrix:Bin size is not homogeneous.                                       Median 49997\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 3. [CD8 T-Cell2]\n",
      "WARNING:hicmatrix.HiCMatrix:Bin size is not homogeneous.                                       Median 49998\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 4. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:initialize 5. [H1 Primed CTCF loops]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t141974420\t141977275\tchr7_KI270803v1_alt\t421393\t422663\t0.943\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142303376\t142304287\tchr7_KI270803v1_alt\t421393\t422663\t0.892\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t340977\t341009\t0.998\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t374336\t375839\t0.987\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t499180\t500347\t0.912\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t340977\t341009\tchr7\t142782517\t142784203\t0.919\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142782517\t142784203\t0.981\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142812722\t142814024\t0.974\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143001731\t143004484\t0.9\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143214492\t143216079\t0.923\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t368583\t369846\tchr7\t142632039\t142633527\t0.878\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t374336\t375839\tchr7\t143007418\t143008973\t0.921\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t403791\t404916\tchr7\t143001731\t143004484\t0.913\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t421393\t422663\tchr7\t142782517\t142784203\t0.986\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t142782517\t142784203\t0.897\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t143214492\t143216079\t0.905\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2086999\t2087770\tchr8_KI270821v1_alt\t548587\t548592\t0.965\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2266587\t2267522\tchr8_KI270821v1_alt\t542877\t544402\t0.999\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t542877\t544402\t0.947\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t548587\t548592\t0.919\n",
      "\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 6. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 7. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 8. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "100%|##########| 323581/323581 [00:22<00:00, 14125.62it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 9. [COPD GWAS]\n",
      "100%|##########| 5492/5492 [00:00<00:00, 20416.03it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 10. [IPF GWAS]\n",
      "100%|##########| 3591/3591 [00:00<00:00, 19895.99it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 11. [CoVid-19_susceptibility]\n",
      "100%|##########| 41/41 [00:00<00:00, 24929.90it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 12. [Ellinghaus CoVid GWAS]\n",
      "100%|##########| 60/60 [00:00<00:00, 24887.09it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 13. [LINE elements]\n",
      "100%|##########| 13418/13418 [00:00<00:00, 19560.89it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 14. [Gencode genes]\n",
      "100%|##########| 108610/108610 [00:08<00:00, 12996.86it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 15. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 16. [CD8_T_mem_rnaseq_minus]\n",
      "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
      "INFO:pygenometracks.tracksClass:103.01595735549927\n",
      "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 50.93085106382979. Dpi is set to 72\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:plotting 2. [CD8 T-Cell1]\n",
      "INFO:pygenometracks.tracks.GenomeTrack:setting min, max values for track 2. [CD8 T-Cell1] to: 0.2, 5.2\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 3. [CD8 T-Cell2]\n",
      "INFO:pygenometracks.tracks.GenomeTrack:setting min, max values for track 3. [CD8 T-Cell2] to: 0.2, 5.2\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 4. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:plotting 5. [H1 Primed CTCF loops]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:120 were links plotted\n",
      "INFO:pygenometracks.tracksClass:plotting 6. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 7. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 8. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 9. [COPD GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning* No intervals were found for file /input_dir/corona_analysis/tracks/COPD_GWAS_hg38.bed in section '9. [COPD GWAS]' for the interval plotted (chr14:96744521-100430762).\n",
      "\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 10. [IPF GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr14 nor 14 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 11. [CoVid-19_susceptibility]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr14 nor 14 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 12. [Ellinghaus CoVid GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr14 nor 14 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 13. [LINE elements]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 3.38,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 14. [Gencode genes]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 23.0,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 15. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 16. [CD8_T_mem_rnaseq_minus]\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr14:96744521-100430762 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/BCL11b_cd8t_hg38_z3.pdf   \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3p Loci - CXCR6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45938415-45955927 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_cxcr6_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45792952-45959481 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_suscept_fine_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45864052-45873481 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_CRE_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45825952-45875481 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_superfine_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr3:45617381-46677708 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_suscept_broad_hg38.pdf   \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr3:44460745-46446068 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_suscept_superbroad_hg38.pdf   \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSTCD \n",
    "\n",
    "* chr4:105662395-105894379 # fine hg38\n",
    "* chr4:105698396-105727514 # tss hg38\n",
    "* chr4:105068288-106695910 # broad hg38\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pygenometracks.tracksClass:initialize 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:initialize 2. [H1 Primed CTCF loops]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t141974420\t141977275\tchr7_KI270803v1_alt\t421393\t422663\t0.943\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142303376\t142304287\tchr7_KI270803v1_alt\t421393\t422663\t0.892\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t340977\t341009\t0.998\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t374336\t375839\t0.987\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t499180\t500347\t0.912\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t340977\t341009\tchr7\t142782517\t142784203\t0.919\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142782517\t142784203\t0.981\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142812722\t142814024\t0.974\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143001731\t143004484\t0.9\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143214492\t143216079\t0.923\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t368583\t369846\tchr7\t142632039\t142633527\t0.878\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t374336\t375839\tchr7\t143007418\t143008973\t0.921\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t403791\t404916\tchr7\t143001731\t143004484\t0.913\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t421393\t422663\tchr7\t142782517\t142784203\t0.986\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t142782517\t142784203\t0.897\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t143214492\t143216079\t0.905\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2086999\t2087770\tchr8_KI270821v1_alt\t548587\t548592\t0.965\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2266587\t2267522\tchr8_KI270821v1_alt\t542877\t544402\t0.999\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t542877\t544402\t0.947\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t548587\t548592\t0.919\n",
      "\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 3. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:initialize 4. [CD8_effmemT_CPG_meth]\n",
      "INFO:pygenometracks.tracksClass:initialize 5. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 6. [T-cell CD8 h3k4me3]\n",
      "INFO:pygenometracks.tracksClass:initialize 7. [T-cell CD8 H3k27Ac]\n",
      "INFO:pygenometracks.tracksClass:initialize 8. [T-cell CD8 H3k9Me3]\n",
      "INFO:pygenometracks.tracksClass:initialize 9. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 10. [T-Cell 2 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 11. [T-Cell 3 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 12. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 13. [T-Cell CM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 14. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "100%|##########| 323581/323581 [00:23<00:00, 14050.36it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 15. [CD8_NAIVE_PRIMARY_CELLS]\n",
      "100%|##########| 350203/350203 [00:24<00:00, 14025.01it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 16. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 17. [COPD GWAS]\n",
      "100%|##########| 5492/5492 [00:00<00:00, 20593.42it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 18. [IPF GWAS]\n",
      "100%|##########| 3591/3591 [00:00<00:00, 19665.14it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 19. [CoVid-19_susceptibility]\n",
      "100%|##########| 41/41 [00:00<00:00, 25393.75it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 20. [Ellinghaus CoVid GWAS]\n",
      "100%|##########| 60/60 [00:00<00:00, 23763.76it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 21. [LINE elements]\n",
      "100%|##########| 13418/13418 [00:00<00:00, 19651.50it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 22. [Gencode genes]\n",
      "100%|##########| 108610/108610 [00:07<00:00, 14392.92it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 23. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 24. [CD8_T_mem_rnaseq_minus]\n",
      "INFO:pygenometracks.tracksClass:initialize 25. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 26. [BALF_control_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 27. [BALF_control_minus]\n",
      "INFO:pygenometracks.tracksClass:initialize 28. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 29. [BALF_infected_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 30. [BALF_infected_minus]\n",
      "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
      "INFO:pygenometracks.tracksClass:62.02659201622009\n",
      "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 56.78191489361703. Dpi is set to 72\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:plotting 2. [H1 Primed CTCF loops]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:1 were links plotted\n",
      "INFO:pygenometracks.tracksClass:plotting 3. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:plotting 4. [CD8_effmemT_CPG_meth]\n",
      "INFO:pygenometracks.tracksClass:plotting 5. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 6. [T-cell CD8 h3k4me3]\n",
      "INFO:pygenometracks.tracksClass:plotting 7. [T-cell CD8 H3k27Ac]\n",
      "INFO:pygenometracks.tracksClass:plotting 8. [T-cell CD8 H3k9Me3]\n",
      "INFO:pygenometracks.tracksClass:plotting 9. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 10. [T-Cell 2 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 11. [T-Cell 3 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 12. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 13. [T-Cell CM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 14. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 15. [CD8_NAIVE_PRIMARY_CELLS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 16. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 17. [COPD GWAS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 3.38,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 18. [IPF GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning* No intervals were found for file /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed in section '18. [IPF GWAS]' for the interval plotted (chr4:105698396-105727514).\n",
      "\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 19. [CoVid-19_susceptibility]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr4 nor 4 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 20. [Ellinghaus CoVid GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr4 nor 4 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 21. [LINE elements]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning* No intervals were found for file /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed in section '21. [LINE elements]' for the interval plotted (chr4:105698396-105727514).\n",
      "\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 22. [Gencode genes]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 23.0,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 23. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 24. [CD8_T_mem_rnaseq_minus]\n",
      "INFO:pygenometracks.tracksClass:plotting 25. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 26. [BALF_control_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 27. [BALF_control_minus]\n",
      "INFO:pygenometracks.tracksClass:plotting 28. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 29. [BALF_infected_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 30. [BALF_infected_minus]\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr4:105698396-105727514 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_TSS_hg38_cd8T.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pygenometracks.tracksClass:initialize 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:initialize 2. [H1 Primed CTCF loops]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t141974420\t141977275\tchr7_KI270803v1_alt\t421393\t422663\t0.943\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142303376\t142304287\tchr7_KI270803v1_alt\t421393\t422663\t0.892\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t340977\t341009\t0.998\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t374336\t375839\t0.987\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t499180\t500347\t0.912\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t340977\t341009\tchr7\t142782517\t142784203\t0.919\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142782517\t142784203\t0.981\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142812722\t142814024\t0.974\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143001731\t143004484\t0.9\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143214492\t143216079\t0.923\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t368583\t369846\tchr7\t142632039\t142633527\t0.878\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t374336\t375839\tchr7\t143007418\t143008973\t0.921\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t403791\t404916\tchr7\t143001731\t143004484\t0.913\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t421393\t422663\tchr7\t142782517\t142784203\t0.986\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t142782517\t142784203\t0.897\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t143214492\t143216079\t0.905\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2086999\t2087770\tchr8_KI270821v1_alt\t548587\t548592\t0.965\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2266587\t2267522\tchr8_KI270821v1_alt\t542877\t544402\t0.999\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t542877\t544402\t0.947\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t548587\t548592\t0.919\n",
      "\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 3. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:initialize 4. [CD8_effmemT_CPG_meth]\n",
      "INFO:pygenometracks.tracksClass:initialize 5. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 6. [T-cell CD8 h3k4me3]\n",
      "INFO:pygenometracks.tracksClass:initialize 7. [T-cell CD8 H3k27Ac]\n",
      "INFO:pygenometracks.tracksClass:initialize 8. [T-cell CD8 H3k9Me3]\n",
      "INFO:pygenometracks.tracksClass:initialize 9. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 10. [T-Cell 2 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 11. [T-Cell 3 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 12. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 13. [T-Cell CM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 14. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "100%|##########| 323581/323581 [00:21<00:00, 14714.56it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 15. [CD8_NAIVE_PRIMARY_CELLS]\n",
      "100%|##########| 350203/350203 [00:23<00:00, 14593.20it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 16. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 17. [COPD GWAS]\n",
      "100%|##########| 5492/5492 [00:00<00:00, 21346.84it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 18. [IPF GWAS]\n",
      "100%|##########| 3591/3591 [00:00<00:00, 20577.70it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 19. [CoVid-19_susceptibility]\n",
      "100%|##########| 41/41 [00:00<00:00, 24507.12it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 20. [Ellinghaus CoVid GWAS]\n",
      "100%|##########| 60/60 [00:00<00:00, 25338.12it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 21. [LINE elements]\n",
      "100%|##########| 13418/13418 [00:00<00:00, 20377.01it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 22. [Gencode genes]\n",
      "100%|##########| 108610/108610 [00:07<00:00, 14811.43it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 23. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 24. [CD8_T_mem_rnaseq_minus]\n",
      "INFO:pygenometracks.tracksClass:initialize 25. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 26. [BALF_control_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 27. [BALF_control_minus]\n",
      "INFO:pygenometracks.tracksClass:initialize 28. [spacer]\n",
      "INFO:pygenometracks.tracksClass:initialize 29. [BALF_infected_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 30. [BALF_infected_minus]\n",
      "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
      "INFO:pygenometracks.tracksClass:59.623852491378784\n",
      "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 56.78191489361703. Dpi is set to 72\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:plotting 2. [H1 Primed CTCF loops]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:3 were links plotted\n",
      "INFO:pygenometracks.tracksClass:plotting 3. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:plotting 4. [CD8_effmemT_CPG_meth]\n",
      "INFO:pygenometracks.tracksClass:plotting 5. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 6. [T-cell CD8 h3k4me3]\n",
      "INFO:pygenometracks.tracksClass:plotting 7. [T-cell CD8 H3k27Ac]\n",
      "INFO:pygenometracks.tracksClass:plotting 8. [T-cell CD8 H3k9Me3]\n",
      "INFO:pygenometracks.tracksClass:plotting 9. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 10. [T-Cell 2 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 11. [T-Cell 3 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 12. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 13. [T-Cell CM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 14. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 15. [CD8_NAIVE_PRIMARY_CELLS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 16. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 17. [COPD GWAS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 14.879999999999999,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 18. [IPF GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning* No intervals were found for file /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed in section '18. [IPF GWAS]' for the interval plotted (chr4:105662395-105894379).\n",
      "\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 19. [CoVid-19_susceptibility]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr4 nor 4 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 20. [Ellinghaus CoVid GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr4 nor 4 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 21. [LINE elements]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 5.68,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 22. [Gencode genes]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 23.0,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 23. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 24. [CD8_T_mem_rnaseq_minus]\n",
      "INFO:pygenometracks.tracksClass:plotting 25. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 26. [BALF_control_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 27. [BALF_control_minus]\n",
      "INFO:pygenometracks.tracksClass:plotting 28. [spacer]\n",
      "INFO:pygenometracks.tracksClass:plotting 29. [BALF_infected_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 30. [BALF_infected_minus]\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr4:105662395-105894379 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_gene_hg38_cd8T.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:pygenometracks.tracksClass:In section 5. [H1 Primed CTCF loops], these parameters are unused:['compact_level']\n",
      "INFO:pygenometracks.tracksClass:initialize 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:initialize 2. [CD8 T-Cell1]\n",
      "WARNING:hicmatrix.HiCMatrix:Bin size is not homogeneous.                                       Median 49997\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 3. [CD8 T-Cell2]\n",
      "WARNING:hicmatrix.HiCMatrix:Bin size is not homogeneous.                                       Median 49998\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 4. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:initialize 5. [H1 Primed CTCF loops]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t141974420\t141977275\tchr7_KI270803v1_alt\t421393\t422663\t0.943\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142303376\t142304287\tchr7_KI270803v1_alt\t421393\t422663\t0.892\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t340977\t341009\t0.998\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t374336\t375839\t0.987\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7\t142338172\t142339623\tchr7_KI270803v1_alt\t499180\t500347\t0.912\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t340977\t341009\tchr7\t142782517\t142784203\t0.919\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142782517\t142784203\t0.981\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t142812722\t142814024\t0.974\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143001731\t143004484\t0.9\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t359102\t359255\tchr7\t143214492\t143216079\t0.923\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t368583\t369846\tchr7\t142632039\t142633527\t0.878\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t374336\t375839\tchr7\t143007418\t143008973\t0.921\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t403791\t404916\tchr7\t143001731\t143004484\t0.913\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t421393\t422663\tchr7\t142782517\t142784203\t0.986\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t142782517\t142784203\t0.897\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr7_KI270803v1_alt\t428100\t429398\tchr7\t143214492\t143216079\t0.905\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2086999\t2087770\tchr8_KI270821v1_alt\t548587\t548592\t0.965\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2266587\t2267522\tchr8_KI270821v1_alt\t542877\t544402\t0.999\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t542877\t544402\t0.947\n",
      "\n",
      "\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:Only links in same chromosome are used. Skipping line\n",
      "chr8\t2294945\t2296283\tchr8_KI270821v1_alt\t548587\t548592\t0.919\n",
      "\n",
      "\n",
      "INFO:pygenometracks.tracksClass:initialize 6. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 7. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:initialize 8. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "100%|##########| 323581/323581 [00:22<00:00, 14441.13it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 9. [COPD GWAS]\n",
      "100%|##########| 5492/5492 [00:00<00:00, 20506.28it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 10. [IPF GWAS]\n",
      "100%|##########| 3591/3591 [00:00<00:00, 20104.90it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 11. [CoVid-19_susceptibility]\n",
      "100%|##########| 41/41 [00:00<00:00, 24430.52it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 12. [Ellinghaus CoVid GWAS]\n",
      "100%|##########| 60/60 [00:00<00:00, 24943.82it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 13. [LINE elements]\n",
      "100%|##########| 13418/13418 [00:00<00:00, 19622.81it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 14. [Gencode genes]\n",
      "100%|##########| 108610/108610 [00:08<00:00, 13159.06it/s]\n",
      "INFO:pygenometracks.tracksClass:initialize 15. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:initialize 16. [CD8_T_mem_rnaseq_minus]\n",
      "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
      "INFO:pygenometracks.tracksClass:102.73052930831909\n",
      "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 50.93085106382979. Dpi is set to 72\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
      "INFO:pygenometracks.tracksClass:plotting 2. [CD8 T-Cell1]\n",
      "WARNING:pygenometracks.tracks.HiCMatrixTrack:The depth was set to 3000000 which is more than 125% of the region plotted. The depth will be set to 2611027\n",
      "INFO:pygenometracks.tracks.GenomeTrack:setting min, max values for track 2. [CD8 T-Cell1] to: 0.2, 5.2\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 3. [CD8 T-Cell2]\n",
      "WARNING:pygenometracks.tracks.HiCMatrixTrack:The depth was set to 3000000 which is more than 125% of the region plotted. The depth will be set to 2611027\n",
      "INFO:pygenometracks.tracks.GenomeTrack:setting min, max values for track 3. [CD8 T-Cell2] to: 0.2, 5.2\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 4. [ENCFF518JVS_cd8_tcell_imputed]\n",
      "INFO:pygenometracks.tracksClass:plotting 5. [H1 Primed CTCF loops]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:23 were links plotted\n",
      "INFO:pygenometracks.tracksClass:plotting 6. [T-Cell 1 cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 7. [T-Cell EM cluster scATAC-seq]\n",
      "INFO:pygenometracks.tracksClass:plotting 8. [CD8_MEMORY_PRIMARY_CELLS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 9. [COPD GWAS]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 58.57999999999999,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 10. [IPF GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning* No intervals were found for file /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed in section '10. [IPF GWAS]' for the interval plotted (chr4:104211178-106300000).\n",
      "\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 1.08,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 11. [CoVid-19_susceptibility]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr4 nor 4 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 12. [Ellinghaus CoVid GWAS]\n",
      "WARNING:pygenometracks.tracks.GenomeTrack:*Warning*\n",
      "Neither chr4 nor 4 existss as a chromosome name inside the bed file. This will generate an empty track!!\n",
      "\n",
      "INFO:pygenometracks.tracksClass:plotting 13. [LINE elements]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 10.28,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 14. [Gencode genes]\n",
      "DEBUG:pygenometracks.tracks.GenomeTrack:ylim 23.0,-0.08\n",
      "INFO:pygenometracks.tracksClass:plotting 15. [CD8_T_mem_rnaseq_plus]\n",
      "INFO:pygenometracks.tracksClass:plotting 16. [CD8_T_mem_rnaseq_minus]\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr4:104211178-106300000 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_broad_hg38_cd8T.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
