{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_lung_fine.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra_hg38.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "ylim = 300000\n",
    "compact_arcs_level = 2\n",
    "line_style = solid\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "#[CTCF NN variants]\n",
    "#file = \n",
    "#title = Variants CTCF perturbing\n",
    "#height = 1\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#labels = false\n",
    "#file_type = bed\n",
    "\n",
    "[ENCFF117BRM_CTCF_NHLF]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF117BRM_CTCF_NHLF.bw\n",
    "title = CTCF chip-seq\n",
    "height = 2\n",
    "color = #42f44e\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[ENCFF782BOR_H3k4Me3_SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF782BOR_H3k4Me3_SAEC.bw\n",
    "title = H3k4Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #ffb42b\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF637KNN_H3k27Ac_NHLF]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF637KNN_H3k27Ac_NHLF.bw\n",
    "title = H3k27Ac fc Chip-seq\n",
    "height = 2\n",
    "color = #42f4f4\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF143LCY_H3k9Me3_NHLF]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF143LCY_H3k9Me3_NHLF.bw\n",
    "title = H3k9Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #a22eea\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[GSM1821904_SC_ATRX_K562]\n",
    "file = /input_dir/corona_analysis/tracks/GSM1821904_hg38-K562-ATRX_SC-enrichment_over_input.bw\n",
    "title = ATRX fc K562 chip-seq SC\n",
    "height = 2\n",
    "color = #9e0afa\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[GSM1821904_Abcam_ATRX_K562]\n",
    "file = /input_dir/corona_analysis/tracks/GSM1821905_hg38-K562-ATRX_Abcam-enrichment_over_input.bw\n",
    "title = ATRX fc K562 chip-seq Abcam\n",
    "height = 2\n",
    "color = #9e0afa\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[NHLF Chromatin state]\n",
    "file = /input_dir/corona_analysis/tracks/E128_15_coreMarks_dense.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = NHLF Chromatin State\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[DNAse Peaks SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF827VFY_DNAse_SAEC.bed\n",
    "title = DNAse Peaks SAEC\n",
    "file_type = narrow_peak\n",
    "height = 1\n",
    "line_width = 1\n",
    "labels = false\n",
    "show_labels = false\n",
    "color = #1d6ae5\n",
    "display = collapsed\n",
    "fontsize = 10\n",
    "\n",
    "[ENCFF827VFY_DNAse_SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF827VFY_DNAse_SAEC.bw\n",
    "title = DNase sensitivity\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Fetal_atlas_ATAC_vasc_endo]\n",
    "file = /input_dir/corona_analysis/tracks/lung_vascular_endothelial_cells_hg38.bw\n",
    "title = Agreggated scATAC-seq lung endothelial\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Fetal_atlas_ATAC_bronch_alveo]\n",
    "file = /input_dir/corona_analysis/tracks/lung_bronchiolar_and_alveolar_epithelial_cells_hg38.bw\n",
    "title = Agreggated scATAC-seq broncheolar and epithelial\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "\n",
    "#rLoop stuff\n",
    "\n",
    "\n",
    "\n",
    "#G-quadruplex stuff\n",
    "[G-quadruplex 93T449]\n",
    "file = /input_dir/corona_analysis/tracks/GSE145535_RAW/GSM4320548_ChIPseq_IP_BG4_rep2_93T449_m1.ucsc.bigWig\n",
    "second_file = /input_dir/corona_analysis/tracks/GSE145535_RAW/GSM4320549_ChIPseq_Input_BG4_rep2_93T449_m1.ucsc.bigWig\n",
    "title = G-quadruplex Liposarcoma (93T449)\n",
    "operation = log2((1 + file) / (1 + second_file))\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[G-quadruplex HaCat]\n",
    "file = /input_dir/corona_analysis/tracks/GSE99205_common_HaCaT_G4_ChIP_peaks_RNase_treated_hg38.bed\n",
    "height = 1\n",
    "title = G-quadruplex HaCat peaks\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "#[Mckay SCLC GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/mckay_2017_sclc_overall_hg38.bed\n",
    "#height = 1\n",
    "#title = Mckay SCLC GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[Wu SCLC GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/gwas_catalog_sclc_hg38.bed\n",
    "#height = 1\n",
    "#title = Wu SCLC GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[COPD GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg38.bed\n",
    "#height = 1\n",
    "#title = COPD GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[IPF GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed\n",
    "#height = 1\n",
    "#title = IPF GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[CoVid-19_susceptibility]\n",
    "#file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "#height = 1\n",
    "#title = CoVid-19 susceptibility 1e-6\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[Ellinghaus CoVid GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "#height = 1\n",
    "#title = Ellinghaus CoVid GWAS\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#file_type = bed\n",
    "\n",
    "[Knowles CF severity]\n",
    "file = /input_dir/corona_analysis/tracks/hg38_CF_severity_knowles_2015_GWAS.bed\n",
    "height = 1\n",
    "title = Knowles Cystic Fibrosis severity GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "#[ClinVar kidney phenotype]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/clinvar_kidney_variants_hg38.bed\n",
    "#height = 1\n",
    "#title = ClinVar w/ kidney phenotype\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#file_type = bed\n",
    "#fontsize = 10\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "max_labels = 0\n",
    "\n",
    "[DTI sgRNAs]\n",
    "file = /input_dir/corona_analysis/tracks/DTI_plasmids/DTI_all_plasmids_hg38.bed\n",
    "height = 2\n",
    "title = DTI sgRNAs\n",
    "fontsize = 7\n",
    "file_type = bed\n",
    "max_labels = 1000\n",
    "color = #0dfff3\n",
    "\n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "#[ENCFF730XSB_HBEC_rnaseq_plus]\n",
    "#file = /input_dir/corona_analysis/tracks/NCFF322WTN_HBEC_rnaseq_plus.bw\n",
    "#title = RNA HBEC +\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "#\n",
    "#[ENCFF730XSB_HBEC_rnaseq_minus]\n",
    "#file = /input_dir/corona_analysis/tracks/ENCFF730XSB_HBEC_rnaseq_minus.bw\n",
    "#title = RNA HBEC -\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "##overlay_previous = share-y\n",
    "#orientation = inverted\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF996XPQ_NHLF_rnaseq_plus_hg38.bw\n",
    "title = RNA NHLF +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF121EOD_NHLF_rnaseq_minus_hg38.bw\n",
    "title = RNA NHLF -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_control_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF control +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_control_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF control -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_infected_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF CoVid +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_infected_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF CoVid -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "#overlay_previous = share-y\n",
    "show_data_range = true\n",
    "file_type = bigwig\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_lung_broad.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[IMR90 In-situ Hi-C]\n",
    "file=/input_dir/corona_analysis/tracks/4DNFIH7TH4MF_50kb_50000.cool\n",
    "title = IMR90 In-situ Hi-C\n",
    "colormap = coolwarm\n",
    "min_value = 1.2\n",
    "max_value = 8\n",
    "depth = 3000000\n",
    "transform = log\n",
    "show_masked_bins = False\n",
    "file_type = hic_matrix\n",
    "height = 10\n",
    "\n",
    "#[TAD Calls]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/HEK293T_100kbTADs_domains.bed\n",
    "#file_type = domains\n",
    "#overlay_previous = share-y\n",
    "#fontsize = 8\n",
    "#line_width = 3\n",
    "#border_color = black\n",
    "#color = none\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra_hg38.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "line_style = solid\n",
    "orientation=inverted\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "[ENCFF827VFY_DNAse_SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF827VFY_DNAse_SAEC.bw\n",
    "title = DNase sensitivity\n",
    "height = 3\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = True\n",
    "file_type = bigwig\n",
    "\n",
    "[GSM1821904_SC_ATRX_K562]\n",
    "file = /input_dir/corona_analysis/tracks/GSM1821904_hg38-K562-ATRX_SC-enrichment_over_input.bw\n",
    "title = ATRX fc K562 chip-seq SC\n",
    "height = 2\n",
    "color = #77a3bd\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[GSM1821904_Abcam_ATRX_K562]\n",
    "file = /input_dir/corona_analysis/tracks/GSM1821905_hg38-K562-ATRX_Abcam-enrichment_over_input.bw\n",
    "title = ATRX fc K562 chip-seq Abcam\n",
    "height = 2\n",
    "color = #77a3bd\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[NHLF Chromatin state]\n",
    "file = /input_dir/corona_analysis/tracks/E128_15_coreMarks_dense.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = NHLF Chromatin State\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "#[Mckay SCLC GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/mckay_2017_sclc_overall_hg38.bed\n",
    "#height = 1\n",
    "#title = Mckay SCLC GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[Wu SCLC GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/gwas_catalog_sclc_hg38.bed\n",
    "#height = 1\n",
    "#title = Wu SCLC GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[COPD GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg38.bed\n",
    "#height = 1\n",
    "#title = COPD GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[IPF GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed\n",
    "#height = 1\n",
    "#title = IPF GWAS\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[CoVid-19_susceptibility]\n",
    "#file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "#height = 1\n",
    "#title = CoVid-19 susceptibility 1e-6\n",
    "#max_labels = 0\n",
    "#color = #e90ef4\n",
    "#file_type = bed\n",
    "#\n",
    "#[Ellinghaus CoVid GWAS]\n",
    "#file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "#height = 1\n",
    "#title = Ellinghaus CoVid GWAS\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#file_type = bed\n",
    "\n",
    "[Knowles CF severity]\n",
    "file = /input_dir/corona_analysis/tracks/hg38_CF_severity_knowles_2015_GWAS.bed\n",
    "height = 1\n",
    "title = Knowles Cystic Fibrosis severity GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "max_labels = 0\n",
    "\n",
    "[tRNA ScanSE]\n",
    "file = /input_dir/corona_analysis/annotations/hg38-tRNAs.bed\n",
    "height = 1\n",
    "title = tRNAs\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "max_labels = 0\n",
    "\n",
    "[DTI sgRNAs]\n",
    "file = /input_dir/corona_analysis/tracks/DTI_plasmids/DTI_all_plasmids_hg38.bed\n",
    "height = 2\n",
    "title = DTI sgRNAs\n",
    "fontsize = 7\n",
    "file_type = bed\n",
    "max_labels = 0\n",
    "color = #0dfff3\n",
    "\n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "#[ENCFF730XSB_HBEC_rnaseq_plus]\n",
    "#file = /input_dir/corona_analysis/tracks/NCFF322WTN_HBEC_rnaseq_plus.bw\n",
    "#title = RNA HBEC +\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "#\n",
    "#[ENCFF730XSB_HBEC_rnaseq_minus]\n",
    "#file = /input_dir/corona_analysis/tracks/ENCFF730XSB_HBEC_rnaseq_minus.bw\n",
    "#title = RNA HBEC -\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "##overlay_previous = share-y\n",
    "#orientation = inverted\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF996XPQ_NHLF_rnaseq_plus_hg38.bw\n",
    "title = RNA NHLF +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF121EOD_NHLF_rnaseq_minus_hg38.bw\n",
    "title = RNA NHLF -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "#[eQTL arcs]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/eQTL_links.arcs\n",
    "#title = eQTL arcs\n",
    "#color = #fc032c\n",
    "#links_type = arcs\n",
    "#line_style = solid\n",
    "#line_width = 1\n",
    "#height = 4\n",
    "#orientation = inverted\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All GWAS regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Visualize all regions significantly correlated w/ CoVid susceptibility\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --BED \"/input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility_50kbMerged.bed\" \\\n",
    "    --outFileName \"/input_dir/corona_analysis/out_viz/CoVid_susceptibility_mphage_finepadded.pdf\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Visualize all regions significantly correlated w/ CoVid susceptibility\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --BED \"/input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility_500kbMerged.bed\" \\\n",
    "    --outFileName \"/input_dir/corona_analysis/out_viz/CoVid_susceptibility_broadpadded_mphage.pdf\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CXCL16"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr17:4738322-4740133 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr17:4732858-4741164 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr17:4403159-5108677 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_TAD_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HLA region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr6:25550790-33630955 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/HLA_TAD_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Histone Clusters\n",
    "\n",
    "chr1:143320028-149902294 -- Histone cluster w/ bunch of tRNAs\n",
    "\n",
    "chr1:161419636-161481412 -- tRNA cluster w/ viral cassette like distribution\n",
    "\n",
    "chr1:228420413-228657700 -- Histone cluster w/ 5s rRNA\n",
    "\n",
    "chr6:25865848-29288735 -- Histone cluster w/ HLA\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr1:143320028-149902294 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Histone_cluster1_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr1:161419636-161481412 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Histone_cluster2_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr1:228420413-228657700 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Histone_cluster3_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr6:25865848-29288735 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Histone_cluster4_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IL6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr7:22725324-22728538 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr7:22724543-22743827 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr7:22491552-23559995 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_TAD_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSTCD \n",
    "\n",
    "* chr4:105662395-105894379 # fine hg38\n",
    "* chr4:105698396-105727514 # tss hg38\n",
    "* chr4:105068288-106695910 # broad hg38\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr4:105698396-105727514 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr4:105662395-105894379 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr4:104211178-106300000 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_broad_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Small Cell Lung Carcinoma regions \n",
    "\n",
    "* MYC\n",
    "* MYCN\n",
    "* TP53\n",
    "* TP73\n",
    "* RB1 (?)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "## Do visualization of fine regions which are present in both datasets\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --BED /input_dir/corona_analysis/tracks/mckay_merge_hg38_sclc.bed \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/SCLC_fine_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "## Do visualization of wide regions which are present in both datasets\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --BED /input_dir/corona_analysis/tracks/mckay_merge_wide_hg38_sclc.bed \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/SCLC_wide_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Myc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr8:127735065-127736548 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/MYC_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr8:127734661-127742944 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/MYC_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr8:127724511-127896163 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/MYC_gene_broad_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr8:126662627-129870643 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/MYC_broad_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP73\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr1:3685628-3698979 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/TP73_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr1:3649699-3749625 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/TP73_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr1:2958490-4256820 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/TP73_broad_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr16:53643753-55367972 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IRX3_FTO_broad_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## KRAS\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr12:25249898-25251819 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/KRAS_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr12:25202758-25253558 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/KRAS_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr12:24833044-25624598 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/KRAS_broad_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CFTR -- Cystic fibrosis\n",
    "\n",
    "Couple of zooms, focusing around L1 elements close to CFTR and some GWAS regions\n",
    "\n",
    "* CFTR TSS = chr7:117479191-117480641\n",
    "* CFTR gene = chr7:117461354-117672910\n",
    "* CFTR IN = chr7:117018827-117937675\n",
    "* CFTR TAD = chr7:116374746-118211808\n",
    "\n",
    "* MUC4 gene = chr3:195719696-195839286\n",
    "* AGTR2 gene = chrX:116038958-116308684"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chrX:116038958-116308684 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/AGTR2_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr3:195719696-195839286 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/MUC4_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr7:117478495-117482027 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CFTR_TSS_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr7:117461354-117672910 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CFTR_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr7:117018827-117937675 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CFTR_IN_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr7:116374746-118211808 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CFTR_TAD_hg38_lung.pdf   \n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
