{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_Mphage_fine.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "ylim = 300000\n",
    "line_style = solid\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "#[CTCF NN variants]\n",
    "#file = \n",
    "#title = Variants CTCF perturbing\n",
    "#height = 1\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#labels = false\n",
    "#file_type = bed\n",
    "\n",
    "[ENCFF117BRM_CTCF_CD14_mono]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF114WQR_cd14_monocyte_hg19_CTCF.bw\n",
    "title = CTCF chip-seq CD14 Mono\n",
    "height = 2\n",
    "color = #42f44e\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Macrophage Methylation]\n",
    "file=/input_dir/corona_analysis/tracks/S0022I51_macrophage_CPG_meth_hg19.bw\n",
    "title = Macrophage Methylation\n",
    "height = .5\n",
    "color = #21ffff\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Macrophage activ Methylation]\n",
    "file=/input_dir/corona_analysis/tracks/S00BS451_activ_macrophage_CPG_meth_hg19.bw\n",
    "title = Macrophage activated methylation\n",
    "height = .5\n",
    "color = #21ffff\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[Macrophage H3k4me3]\n",
    "file = /input_dir/corona_analysis/tracks/S0022IH1_macrophage_h3k4me3_hg19.bw\n",
    "title = H3k4Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #ffb42b\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Macrophage H3k27Ac]\n",
    "file = /input_dir/corona_analysis/tracks/S0022IH1_macrophage_h3k27Ac_hg19.bw\n",
    "title = H3k27Ac fc Chip-seq\n",
    "height = 2\n",
    "color = #42f4f4\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Macrophage H3k9me3]\n",
    "file = /input_dir/corona_analysis/tracks/S0022IH1_macrophage_h3k9me3_hg19.bw\n",
    "title = H3k9Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #a22eea\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[CD14_mono]\n",
    "file = /input_dir/corona_analysis/tracks/E124_15_coreMarks_dense.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD14 Monocyte Chromatin State\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[DNAse Peaks Macrophage]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I44_macrophage_DNAse_hg19.bed\n",
    "title = DNAse Peaks \n",
    "file_type = bed\n",
    "height = 1\n",
    "line_width = 1\n",
    "labels = false\n",
    "show_labels = false\n",
    "color = #1d6ae5\n",
    "display = collapsed\n",
    "fontsize = 10\n",
    "\n",
    "[Macrophage Accessibility]\n",
    "#file = /input_dir/corona_analysis/tracks/S0022I44_macrophage_DNAse_hg19.bw\n",
    "file = /input_dir/corona_analysis/tracks/Mono_2_scATAC_hg19.bw \n",
    "title = Macrophage (mono 2) scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "#R-loop stuff\n",
    "[GSE145964_ssDRIP-MSC-forward]\n",
    "file = /input_dir/corona_analysis/tracks/GSE145964_ssDRIP-MSC.forward.bw\n",
    "title = R-loop ssDRIP MSC forward strand\n",
    "height = 2\n",
    "color = #f92a87\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[GSE145964_ssDRIP-MSC-reverse]\n",
    "file = /input_dir/corona_analysis/tracks/GSE145964_ssDRIP-MSC.reverse.bw\n",
    "title = R-loop ssDRIP MSC reverse strand\n",
    "height = 2\n",
    "color = #f92a87\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "#G-quadruplex stuff\n",
    "[G-quadruplex 93T449]\n",
    "file = /input_dir/corona_analysis/tracks/GSE145535_RAW/GSM4320548_ChIPseq_IP_BG4_rep2_93T449_m1.bw\n",
    "second_file = /input_dir/corona_analysis/tracks/GSM4320549_ChIPseq_Input_BG4_rep2_93T449_m1_hg19.bw\n",
    "title = G-quadruplex Liposarcoma (93T449)\n",
    "operation = log2((1 + file) / (1 + second_file))\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[G-quadruplex HaCat]\n",
    "file = /input_dir/corona_analysis/tracks/GSE99205_common_HaCaT_G4_ChIP_peaks_RNase_treated_hg19.bed\n",
    "height = 1\n",
    "title = G-quadruplex HaCat peaks\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[COPD GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg19.bed\n",
    "height = 1\n",
    "title = COPD GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[IPF GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/ipf_gwas_hg19_allen_5e-8.bed\n",
    "height = 1\n",
    "title = IPF GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg19_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg19.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "[DTI sgRNAs]\n",
    "file = /input_dir/corona_analysis/tracks/DTI_plasmids/DTI_all_plasmids_hg19.bed\n",
    "height = 2\n",
    "title = DTI sgRNAs\n",
    "fontsize = 7\n",
    "file_type = bed\n",
    "max_labels = 1000\n",
    "color = #0dfff3\n",
    "\n",
    "#[ClinVar kidney phenotype]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/clinvar_kidney_variants_hg19.bed\n",
    "#height = 1\n",
    "#title = ClinVar w/ kidney phenotype\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#file_type = bed\n",
    "#fontsize = 10\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg19_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Repeatmasker]         \n",
    "file = /input_dir/corona_analysis/annotations/human/hg19_2020_rmsk_sorted_filt_col.bed\n",
    "height = 1\n",
    "title = Repeats\n",
    "color = bed_rgb\n",
    "border_color = bed_rgb\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "display = collapsed\n",
    "gene_rows = 4\n",
    "              \n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v37.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "[Mphage_rnaseq_plus]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I12_Mphage_rnaseq_plus_hg19.bw\n",
    "title = RNA Mphage +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Mphage_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I12_Mphage_rnaseq_minus_hg19.bw\n",
    "title = RNA Mphage -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[Mphage_activ_rnaseq_plus]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I12_Mphage_activ_rnaseq_plus_hg19.bw\n",
    "title = RNA Inflammatory Mphage +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Mphage_activ_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I12_Mphage_activ_rnaseq_minus_hg19.bw\n",
    "title = RNA Inflammatory Mphage -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "orientation = inverted\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_Mphage_broad.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[PMA THP1 cell]\n",
    "#file=/input_dir/corona_analysis/tracks/Macrophage_HiC_hg19_50000.cool\n",
    "file=/input_dir/corona_analysis/tracks/A_inter_test_30_50kb.cool\n",
    "#file=/input_dir/corona_analysis/tracks/A_inter_test.cool\n",
    "title = PMA THP1 Hi-C\n",
    "colormap = coolwarm\n",
    "min_value = .2\n",
    "max_value = 8\n",
    "depth = 3000000\n",
    "transform = log\n",
    "show_masked_bins = False\n",
    "file_type = hic_matrix\n",
    "height = 10\n",
    "\n",
    "#[TAD Calls]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/HEK293T_100kbTADs_domains.bed\n",
    "#file_type = domains\n",
    "#overlay_previous = share-y\n",
    "#fontsize = 8\n",
    "#line_width = 3\n",
    "#border_color = black\n",
    "#color = none\n",
    "\n",
    "[ENCFF117BRM_CTCF_CD14_mono]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF114WQR_cd14_monocyte_hg19_CTCF.bw\n",
    "title = CTCF chip-seq CD14 Mono\n",
    "height = 3\n",
    "color = #42f44e\n",
    "#overlay previous = yes\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "max_value = 10\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = True\n",
    "file_type = bigwig\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "line_style = solid\n",
    "orientation=inverted\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "[Macrophage DNAse]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I44_macrophage_DNAse_hg19.bw\n",
    "title = DNase sensitivity\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[G-quadruplex HaCat]\n",
    "file = /input_dir/corona_analysis/tracks/GSE99205_common_HaCaT_G4_ChIP_peaks_RNase_treated_hg19.bed\n",
    "height = 1\n",
    "title = G-quadruplex HaCat peaks\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[CD14_mono]\n",
    "file = /input_dir/corona_analysis/tracks/E124_15_coreMarks_dense.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD14 Monocyte Chromatin State\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[COPD GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg19.bed\n",
    "height = 1\n",
    "title = COPD GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[IPF GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/ipf_gwas_hg19_allen_5e-8.bed\n",
    "height = 1\n",
    "title = IPF GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg19_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg19.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg19_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Repeatmasker]         \n",
    "file = /input_dir/corona_analysis/annotations/human/hg19_2020_rmsk_sorted_filt_col.bed\n",
    "height = 1\n",
    "title = Repeats\n",
    "color = bed_rgb\n",
    "border_color = bed_rgb\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "display = collapsed\n",
    "gene_rows = 4\n",
    "              \n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v37.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "[Mphage_activ_rnaseq_plus]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I12_Mphage_activ_rnaseq_plus_hg19.bw\n",
    "title = RNA Inflammatory Mphage +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[Mphage_activ_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/S0022I12_Mphage_activ_rnaseq_minus_hg19.bw\n",
    "title = RNA Inflammatory Mphage -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "orientation = inverted\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IL6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr7:22764943-22768157 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_TSS_hg19_mphage.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr7:22764162-22783446 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_gene_hg19_mphage.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --region chr7:22531171-23599614 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_TAD_hg19_broad_mphage.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CXCL16 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr17:4641617-4643428 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_TSS_hg19_mphage.pdf   \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr17:4636153-4644459 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_gene_hg19_mphage.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --region chr17:4306454-5011972 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_TAD_hg19_mphage.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSTCD \n",
    "\n",
    "chr4:105989445-107617067 # broad hg19\n",
    "chr4:106583552-106815536 # fine hg19\n",
    "chr4:106619553-106648671 # tss hg19\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr4:106619553-106648671 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_TSS_hg19_mphage.pdf   \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr4:106583552-106815536 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_gene_hg19_mphage.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --region chr4:105132335-107221157 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_TAD_hg19_mphage.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sepsis loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --region chr2:218816503-219597753 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Sepsis_chr2_TAD_hg19_mphage.pdf   \n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CXCR1 and CXCR2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr2:218966728-219046286 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCR1_CXCR2_gene_hg19_mphage.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### IRF8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr16:85919481-85969142 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IRF8_gene_hg19_mphage.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --region chr16:85805972-87761605 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Sepsis_chr16_TAD_hg19_mphage.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FLT1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr13:29067433-29071611 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/FLT1_tss_hg19_mphage.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --region chr13:28873193-29097266 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/FLT1_gene_hg19_mphage.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --region chr13:28856331-29545514 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/FLT1_TAD_hg19_mphage.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Zip output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tar -zcvf out_viz.tar.gz /input_dir/corona_analysis/out_viz"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
