{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_tcell_fine.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra_hg38.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "ylim = 300000\n",
    "line_style = solid\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "#[CTCF NN variants]\n",
    "#file = \n",
    "#title = Variants CTCF perturbing\n",
    "#height = 1\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#labels = false\n",
    "#file_type = bed\n",
    "\n",
    "[ENCFF518JVS_cd8_tcell_imputed]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF518JVS_cd8_tcell_imputed_hg38.bw\n",
    "title = CTCF chip-seq CD8 Tcell imputed\n",
    "height = 3\n",
    "color = #42f44e\n",
    "#overlay previous = yes\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "max_value = 10\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = True\n",
    "file_type = bigwig\n",
    "\n",
    "[CD8_effmemT_CPG_meth]\n",
    "file = /input_dir/corona_analysis/tracks/S002ND51_CD8_effmemT_CPG_meth_hg38.bw\n",
    "title = CD8+ eff memory T-cell C-meth\n",
    "height = .5\n",
    "color = #21ffff\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[T-cell CD8 h3k4me3]\n",
    "file = /input_dir/corona_analysis/tracks/C0054X_cd8_tcell_h3k4me3_hg38.bw\n",
    "title = CD8 T-cell H3k4Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #ffb42b\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-cell CD8 H3k27Ac]\n",
    "file = /input_dir/corona_analysis/tracks/C0054X_cd8_tcell_h3k27Ac_hg38.bw\n",
    "title = CD8 T-cell H3k27Ac fc Chip-seq\n",
    "height = 2\n",
    "color = #42f4f4\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-cell CD8 H3k9Me3]\n",
    "file = /input_dir/corona_analysis/tracks/C0054X_cd8_tcell_h3k9me3_hg38.bw\n",
    "title = CD8 T-cell H3k9Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #a22eea\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell 1 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T1_scATAC_hg38.bw\n",
    "title = [T-Cell 1 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell 2 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T2_scATAC_hg38.bw\n",
    "title = [T-Cell 2 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig       \n",
    "\n",
    "[T-Cell 3 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T3_scATAC_hg38.bw\n",
    "title = T-Cell 3 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell EM cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_EM_T_scATAC_hg38.bw\n",
    "title = T-Cell EM cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell CM cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_CM_T_scATAC_hg38.bw\n",
    "title = T-Cell CM cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         \n",
    "[CD8_MEMORY_PRIMARY_CELLS]\n",
    "#file = /input_dir/corona_analysis/tracks/CD8_MEMORY_PRIMARY_CELLS_hg38.bed.gz\n",
    "file = /input_dir/corona_analysis/tracks/E048_15_coreMarks_dense_hg38.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD8 Tcell memory cells\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[CD8_NAIVE_PRIMARY_CELLS]\n",
    "#file = /input_dir/corona_analysis/tracks/CD8_NAIVE_PRIMARY_CELLS_hg38.bed.gz\n",
    "file = /input_dir/corona_analysis/tracks/E047_15_coreMarks_dense_hg38.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD8 Tcell naive cells\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[G-quadruplex 93T449]\n",
    "file = /input_dir/corona_analysis/tracks/GSE145535_RAW/GSM4320548_ChIPseq_IP_BG4_rep2_93T449_m1.ucsc.bigWig\n",
    "second_file = /input_dir/corona_analysis/tracks/GSE145535_RAW/GSM4320549_ChIPseq_Input_BG4_rep2_93T449_m1.ucsc.bigWig\n",
    "title = G-quadruplex Liposarcoma (93T449)\n",
    "operation = log2((1 + file) / (1 + second_file))\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[G-quadruplex HaCat]\n",
    "file = /input_dir/corona_analysis/tracks/GSE99205_common_HaCaT_G4_ChIP_peaks_RNase_treated_hg38.bed\n",
    "height = 1\n",
    "title = G-quadruplex HaCat peaks\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed        \n",
    "         \n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[Mckay SCLC GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/mckay_2017_sclc_overall_hg38.bed\n",
    "height = 1\n",
    "title = Mckay SCLC GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Wu SCLC GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/gwas_catalog_sclc_hg38.bed\n",
    "height = 1\n",
    "title = Wu SCLC GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed    \n",
    "         \n",
    "[COPD GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg38.bed\n",
    "height = 1\n",
    "title = COPD GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed         \n",
    "         \n",
    "[IPF GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed\n",
    "height = 1\n",
    "title = IPF GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed      \n",
    "         \n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "         \n",
    "[DTI sgRNAs]\n",
    "file = /input_dir/corona_analysis/tracks/DTI_plasmids/DTI_all_plasmids_hg38.bed\n",
    "height = 2\n",
    "title = DTI sgRNAs\n",
    "fontsize = 7\n",
    "file_type = bed\n",
    "max_labels = 1000\n",
    "color = #0dfff3\n",
    "\n",
    "#[ClinVar kidney phenotype]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/clinvar_kidney_variants_hg38.bed\n",
    "#height = 1\n",
    "#title = ClinVar w/ immune phenotype\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#file_type = bed\n",
    "#fontsize = 10\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Repeatmasker]         \n",
    "file = /input_dir/corona_analysis/annotations/human/hg38_2020_rmsk_sorted_filt_col.bed\n",
    "height = 1\n",
    "title = Repeats\n",
    "color = bed_rgb\n",
    "border_color = bed_rgb\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "display = collapsed\n",
    "gene_rows = 4\n",
    "              \n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "[CD8_T_mem_rnaseq_plus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_plus.bw\n",
    "title = RNA CD8 TCell +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         \n",
    "[CD8_T_mem_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_minus.bw\n",
    "title = RNA CD8 TCell -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_control_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF control +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_control_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF control -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_infected_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF CoVid +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_infected_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF CoVid -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "#overlay_previous = share-y\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_tcell_broad.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[CD8 T-Cell1]\n",
    "file=/input_dir/corona_analysis/tracks/GSM2827788_CD8T1_hg_t.cool\n",
    "title = Tcell S1\n",
    "colormap = coolwarm\n",
    "min_value = .2\n",
    "max_value = 5.2\n",
    "depth = 3000000\n",
    "transform = log\n",
    "show_masked_bins = False\n",
    "file_type = hic_matrix\n",
    "height = 10\n",
    "\n",
    "[CD8 T-Cell2]\n",
    "file=/input_dir/corona_analysis/tracks/GSM2827789_CD8T2_hg_t.cool\n",
    "title = Tcell S2\n",
    "colormap = coolwarm\n",
    "min_value = .2\n",
    "max_value = 5.2\n",
    "depth = 3000000\n",
    "transform = log\n",
    "show_masked_bins = False\n",
    "file_type = hic_matrix\n",
    "height = 10\n",
    "\n",
    "#[TAD Calls]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/HEK293T_100kbTADs_domains.bed\n",
    "#file_type = domains\n",
    "#overlay_previous = share-y\n",
    "#fontsize = 8\n",
    "#line_width = 3\n",
    "#border_color = black\n",
    "#color = none\n",
    "\n",
    "[ENCFF518JVS_cd8_tcell_imputed]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF518JVS_cd8_tcell_imputed_hg38.bw\n",
    "title = CTCF chip-seq CD8 Tcell imputed\n",
    "height = 3\n",
    "color = #42f44e\n",
    "#overlay previous = yes\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "max_value = 10\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = True\n",
    "file_type = bigwig\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra_hg38.arcs\n",
    "title = Cohesin Chia-PET in hESC\n",
    "compact_level = 1\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "line_style = solid\n",
    "orientation=inverted\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "[T-Cell 1 cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_N_T1_scATAC_hg38.bw\n",
    "title = [T-Cell 1 cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[T-Cell EM cluster scATAC-seq]\n",
    "file = /input_dir/corona_analysis/tracks/CD8_EM_T_scATAC_hg38.bw\n",
    "title = T-Cell EM cluster scATAC-seq\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[CD8_MEMORY_PRIMARY_CELLS]\n",
    "#file = /input_dir/corona_analysis/tracks/CD8_MEMORY_PRIMARY_CELLS_hg38.bed.gz\n",
    "file = /input_dir/corona_analysis/tracks/E048_15_coreMarks_dense_hg38.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = CD8 Tcell memory cells\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[COPD GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/COPD_GWAS_hg38.bed\n",
    "height = 1\n",
    "title = COPD GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed      \n",
    "         \n",
    "[IPF GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/ipf_gwas_hg38_allen_5e-8.bed\n",
    "height = 1\n",
    "title = IPF GWAS\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "         \n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "         \n",
    "[CD8_T_mem_rnaseq_plus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_plus.bw\n",
    "title = RNA CD8 TCell +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "         \n",
    "[CD8_T_mem_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/C001HN_CD8_T_mem_rnaseq_minus.bw\n",
    "title = RNA CD8 TCell -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IL2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr4:122435776-122467762 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL2_cd8t_hg38_fine.pdf   \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Innate immune"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# T-cell differentiation\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr14:98785652-98836861 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/BCL11b_cd8t_hg38_z1.pdf   \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr14:98669194-99282480 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/BCL11b_cd8t_hg38_z2.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr14:96744521-100430762 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/BCL11b_cd8t_hg38_z3.pdf   \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3p Loci - CXCR6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45938415-45955927 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_cxcr6_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45792952-45959481 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_suscept_fine_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45864052-45873481 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_CRE_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr3:45825952-45875481 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_superfine_hg38.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr3:45617381-46677708 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_suscept_broad_hg38.pdf   \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr3:44460745-46446068 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/chr3p_cd8T_CoVid_suscept_superbroad_hg38.pdf   \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr13:28239868-28624208 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/FLT1_gene_cd8T_hg38.pdf   \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSTCD \n",
    "\n",
    "* chr4:105662395-105894379 # fine hg38\n",
    "* chr4:105698396-105727514 # tss hg38\n",
    "* chr4:105068288-106695910 # broad hg38\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr4:105698396-105727514 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_TSS_hg38_cd8T.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr4:105662395-105894379 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_gene_hg38_cd8T.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr4:104211178-106300000 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/GSTCD_broad_hg38_cd8T.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sepsis loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr2:217951780-218733030 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Sepsis_chr2_TAD_hg38_CD8T.pdf   \n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CXCR1 and CXCR2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr2:218102005-218181563 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCR1_CXCR2_gene_hg38_CD8T.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### IRF8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr16:85885875-85935536 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IRF8_gene_hg38_CD8T.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr16:85772366-87727999 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/Sepsis_chr16_TAD_hg38_CD8T.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FLT1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr13:28493296-28497474 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/FLT1_tss_hg38_CD8T.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_fine.ini \\\n",
    "    --region chr13:28299056-28523129 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/FLT1_gene_hg38_CD8T.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_tcell_broad.ini \\\n",
    "    --region chr13:28282194-28971377 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/FLT1_TAD_hg38_CD8T.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
