{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_lung_fine.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "ylim = 3000000\n",
    "line_style = solid\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "#[CTCF NN variants]\n",
    "#file = \n",
    "#title = Variants CTCF perturbing\n",
    "#height = 1\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#labels = false\n",
    "#file_type = bed\n",
    "\n",
    "[ENCFF117BRM_CTCF_NHLF]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF117BRM_CTCF_NHLF.bw\n",
    "title = CTCF chip-seq\n",
    "height = 2\n",
    "color = #42f44e\n",
    "orientation = inverted\n",
    "min_value = 2\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "#[GSE50874_tubule_avg_methylation]\n",
    "#file=/input_dir/datavol/epigenetics_tracks/GSE50874_tubule_avg_methylation.bw\n",
    "#title = Methylation tubules\n",
    "#height = .5\n",
    "#color = #21ffff\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "#orientation = inverted\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[ENCFF782BOR_H3k4Me3_SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF782BOR_H3k4Me3_SAEC.bw\n",
    "title = H3k4Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #ffb42b\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF637KNN_H3k27Ac_NHLF]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF637KNN_H3k27Ac_NHLF.bw\n",
    "title = H3k27Ac fc Chip-seq\n",
    "height = 2\n",
    "color = #42f4f4\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF143LCY_H3k9Me3_NHLF]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF143LCY_H3k9Me3_NHLF.bw\n",
    "title = H3k9Me3 fc Chip-seq\n",
    "height = 2\n",
    "color = #a22eea\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[NHLF Chromatin state]\n",
    "file = /input_dir/corona_analysis/tracks/E128_15_coreMarks_dense.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = NHLF Chromatin State\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[DNAse Peaks SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF827VFY_DNAse_SAEC.bed\n",
    "title = DNAse Peaks SAEC\n",
    "file_type = narrow_peak\n",
    "height = 1\n",
    "line_width = 1\n",
    "labels = false\n",
    "show_labels = false\n",
    "color = #1d6ae5\n",
    "display = collapsed\n",
    "fontsize = 10\n",
    "\n",
    "[ENCFF827VFY_DNAse_SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF827VFY_DNAse_SAEC.bw\n",
    "title = DNase sensitivity\n",
    "height = 2\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "orientation = inverted\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "#[ClinVar kidney phenotype]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/clinvar_kidney_variants_hg19.bed\n",
    "#height = 1\n",
    "#title = ClinVar w/ kidney phenotype\n",
    "#color = #e90ef4\n",
    "#max_labels = 0\n",
    "#file_type = bed\n",
    "#fontsize = 10\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "#[ENCFF730XSB_HBEC_rnaseq_plus]\n",
    "#file = /input_dir/corona_analysis/tracks/NCFF322WTN_HBEC_rnaseq_plus.bw\n",
    "#title = RNA HBEC +\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "#\n",
    "#[ENCFF730XSB_HBEC_rnaseq_minus]\n",
    "#file = /input_dir/corona_analysis/tracks/ENCFF730XSB_HBEC_rnaseq_minus.bw\n",
    "#title = RNA HBEC -\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "##overlay_previous = share-y\n",
    "#orientation = inverted\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF996XPQ_NHLF_rnaseq_plus_hg38.bw\n",
    "title = RNA NHLF +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF121EOD_NHLF_rnaseq_minus_hg38.bw\n",
    "title = RNA NHLF -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_control_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF control +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_control_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_control_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF control -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[spacer]\n",
    "height = 1\n",
    "\n",
    "[BALF_infected_plus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_minus_hg38.bw\n",
    "title = RNA BALF CoVid +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[BALF_infected_minus]\n",
    "file = /input_dir/corona_analysis/tracks/BALF_corona_s1_rnaseq_plus_hg38.bw\n",
    "title = RNA BALF CoVid -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "orientation = inverted\n",
    "#overlay_previous = share-y\n",
    "show_data_range = true\n",
    "file_type = bigwig\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile regulatory_architecture_lung_broad.ini\n",
    "\n",
    "[x-axis]\n",
    "where = top\n",
    "\n",
    "[IMR90 In-situ Hi-C]\n",
    "file=/input_dir/corona_analysis/tracks/4DNFIH7TH4MF_50kb_50000.cool\n",
    "title = IMR90 In-situ Hi-C\n",
    "colormap = coolwarm\n",
    "min_value = 1.2\n",
    "max_value = 8\n",
    "depth = 3000000\n",
    "transform = log\n",
    "show_masked_bins = False\n",
    "file_type = hic_matrix\n",
    "height = 10\n",
    "\n",
    "#[TAD Calls]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/HEK293T_100kbTADs_domains.bed\n",
    "#file_type = domains\n",
    "#overlay_previous = share-y\n",
    "#fontsize = 8\n",
    "#line_width = 3\n",
    "#border_color = black\n",
    "#color = none\n",
    "\n",
    "[H1 Primed CTCF loops]\n",
    "file = /input_dir/corona_analysis/tracks/primed_.7_origami_intra.arcs\n",
    "title = Cohesin Chia-PET\n",
    "color = #00a110\n",
    "links_type = arcs\n",
    "line_style = solid\n",
    "orientation=inverted\n",
    "line_width = 1.5\n",
    "height = 5\n",
    "\n",
    "[ENCFF827VFY_DNAse_SAEC]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF827VFY_DNAse_SAEC.bw\n",
    "title = DNase sensitivity\n",
    "height = 3\n",
    "color = #2a42f9\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = True\n",
    "file_type = bigwig\n",
    "\n",
    "[NHLF Chromatin state]\n",
    "file = /input_dir/corona_analysis/tracks/E128_15_coreMarks_dense.bed.gz\n",
    "fontsize = 8\n",
    "height = 1\n",
    "title = NHLF Chromatin State\n",
    "display = color from bed rgb\n",
    "labels = false\n",
    "color = bed_rgb\n",
    "display = collapsed\n",
    "\n",
    "[CoVid-19_susceptibility]\n",
    "file = /input_dir/corona_analysis/tracks/covid_hg38_1e6_susceptibility.bed\n",
    "height = 1\n",
    "title = CoVid-19 susceptibility 1e-6\n",
    "max_labels = 0\n",
    "color = #e90ef4\n",
    "file_type = bed\n",
    "\n",
    "[Ellinghaus CoVid GWAS]\n",
    "file = /input_dir/corona_analysis/tracks/Ellinghaus_covid_gwas_hg38.bed\n",
    "height = 1\n",
    "title = Ellinghaus CoVid GWAS\n",
    "color = #e90ef4\n",
    "max_labels = 0\n",
    "file_type = bed\n",
    "\n",
    "[LINE elements]\n",
    "file = /input_dir/corona_analysis/annotations/hsflnil1_hg38_sorted.bed\n",
    "height = 1\n",
    "title = LINE retroelements\n",
    "fontsize = 10\n",
    "file_type = bed\n",
    "\n",
    "[Gencode genes]\n",
    "file = /input_dir/corona_analysis/annotations/gencode_v38.sorted.bed\n",
    "height = 4\n",
    "title = Gencode Genes\n",
    "fontsize = 8\n",
    "file_type = bed\n",
    "display = stacked\n",
    "labels_in_margin = false\n",
    "style = UCSC\n",
    "gene_rows = 10\n",
    "max_labels = 5000000\n",
    "\n",
    "#[ENCFF730XSB_HBEC_rnaseq_plus]\n",
    "#file = /input_dir/corona_analysis/tracks/NCFF322WTN_HBEC_rnaseq_plus.bw\n",
    "#title = RNA HBEC +\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "#\n",
    "#[ENCFF730XSB_HBEC_rnaseq_minus]\n",
    "#file = /input_dir/corona_analysis/tracks/ENCFF730XSB_HBEC_rnaseq_minus.bw\n",
    "#title = RNA HBEC -\n",
    "#height = 2\n",
    "#color = #ef1a1a\n",
    "#min_value = 0\n",
    "#number_of_bins = 500\n",
    "#nans_to_zeros = True\n",
    "#summary_method = mean\n",
    "##overlay_previous = share-y\n",
    "#orientation = inverted\n",
    "#show_data_range = true\n",
    "#file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF996XPQ_NHLF_rnaseq_plus_hg38.bw\n",
    "title = RNA NHLF +\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "[ENCFF121EOD_NHLF_rnaseq_minus]\n",
    "file = /input_dir/corona_analysis/tracks/ENCFF121EOD_NHLF_rnaseq_minus_hg38.bw\n",
    "title = RNA NHLF -\n",
    "height = 2\n",
    "color = #ef1a1a\n",
    "min_value = 0\n",
    "number_of_bins = 500\n",
    "nans_to_zeros = True\n",
    "summary_method = mean\n",
    "#overlay_previous = share-y\n",
    "orientation = inverted\n",
    "show_data_range = true\n",
    "file_type = bigwig\n",
    "\n",
    "#[Suztak eQTL arcs Tubules]\n",
    "#file = /input_dir/datavol/epigenetics_tracks/suzstak_eQTL_Tubules_all.arcs\n",
    "#title = Suztak eQTL arcs Tubules\n",
    "#color = #fc032c\n",
    "#links_type = arcs\n",
    "#line_style = solid\n",
    "#line_width = 1\n",
    "#height = 4\n",
    "#orientation = inverted\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All GWAS regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Visualize all regions significantly correlated w/ CoVid susceptibility\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_fine.ini \\\n",
    "    --BED \"/input_dir/corona_analysis/tracks/covid_hg19_1e6_susceptibility_50kbMerged.bed\" \\\n",
    "    --outFileName \"/input_dir/corona_analysis/out_viz/CoVid_susceptibility_mphage_finepadded.pdf\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Visualize all regions significantly correlated w/ CoVid susceptibility\n",
    "pyGenomeTracks --tracks regulatory_architecture_Mphage_broad.ini \\\n",
    "    --BED \"/input_dir/corona_analysis/tracks/covid_hg19_1e6_susceptibility_500kbMerged.bed\" \\\n",
    "    --outFileName \"/input_dir/corona_analysis/out_viz/CoVid_susceptibility_broadpadded_mphage.pdf\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CXCL16"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr17:4738322-4740133 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr17:4732858-4741164 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr17:4403159-5108677 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/CXCL16_TAD_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IL6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr7:22725324-22728538 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_TSS_hg38_lung.pdf   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_fine.ini \\\n",
    "    --region chr7:22724543-22743827 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_gene_hg38_lung.pdf\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "pyGenomeTracks --tracks regulatory_architecture_lung_broad.ini \\\n",
    "    --region chr7:22491552-23559995 \\\n",
    "    --outFileName /input_dir/corona_analysis/out_viz/IL6_TAD_hg38_lung.pdf   \n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
