{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get reference transcriptome for normal BLFs\n",
    "\n",
    "Transcriptome comes in the form of 3 normal controls RNA-seq BALF sequenced on Illumina Hiseq 2000\n",
    "\n",
    "* SRX7253176 - SRR10571724 Ctrl1\n",
    "* SRX7253170 - SRR10571730 Ctlr2\n",
    "* SRX7253168 - SRR10571732 Ctlr3\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "##prefetch SRA files\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176.sra SRX7253176\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170.sra SRX7253170\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168.sra SRX7253168\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "#Unpack SRA files, toss technicals, zip into fq.gz files\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176.sra \\\n",
    "    --dumpbase --clip --gzip --skip-technical --readids \\\n",
    "    --read-filter pass --split-files \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/BALF_control/\n",
    "\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170.sra \\\n",
    "    --dumpbase --clip --gzip --skip-technical --readids \\\n",
    "    --read-filter pass --split-files \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/BALF_control/\n",
    "\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168.sra \\\n",
    "    --dumpbase --clip --gzip --skip-technical --readids \\\n",
    "    --read-filter pass --split-files \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/BALF_control/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Filter input RNA-seq reads from control BALF lung of 3 donors\n",
    "#Ctrl1\n",
    "fastp -p 14 -i /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176_pass_1.fastq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176_pass_2.fastq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176_filt_1.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176_filt_2.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/BALF_control/SRX7253176_fastp.json \n",
    "\n",
    "#Ctrl2\n",
    "fastp -p 14 -i /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170_pass_1.fastq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170_pass_2.fastq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170_filt_1.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170_filt_2.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/BALF_control/SRX7253170_fastp.json \n",
    "\n",
    "#Ctrl3\n",
    "fastp -p 14 -i /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168_pass_1.fastq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168_pass_2.fastq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168_filt_1.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168_filt_2.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/BALF_control/SRX7253168_fastp.json \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir /input_dir/corona_analysis/alignment_out/BALF_control_quant/\n",
    "\n",
    "#Align reads against human transcriptome for BALF control\n",
    "cd /input_dir/corona_analysis/input_reads/BALF_control\n",
    "#Ctlr1\n",
    "STAR --runMode alignReads      \\\n",
    "     --genomeDir /input_dir/corona_analysis/annotations/human/STAR_ix \\\n",
    "     --outSAMtype BAM SortedByCoordinate --runThreadN 14 \\\n",
    "     --outFileNamePrefix /input_dir/corona_analysis/alignment_out/BALF_control_quant/BALF_control_s1_\\\n",
    "     --outTmpDir /input_dir/corona_analysis/temp/star2 \\\n",
    "     --outReadsUnmapped Fastx --limitBAMsortRAM 27000000000 \\\n",
    "     --outWigType wiggle --quantMode TranscriptomeSAM   --readFilesCommand zcat \\\n",
    "     --readFilesIn SRX7253176_filt_1.fq.gz \\\n",
    "                   SRX7253176_filt_2.fq.gz\n",
    "#Ctlr2\n",
    "STAR --runMode alignReads      \\\n",
    "     --genomeDir /input_dir/corona_analysis/annotations/human/STAR_ix \\\n",
    "     --outSAMtype BAM SortedByCoordinate --runThreadN 14 \\\n",
    "     --outFileNamePrefix /input_dir/corona_analysis/alignment_out/BALF_control_quant/BALF_control_s2_\\\n",
    "     --outTmpDir /input_dir/corona_analysis/temp/star2 \\\n",
    "     --outReadsUnmapped Fastx --limitBAMsortRAM 27000000000 \\\n",
    "     --outWigType wiggle --quantMode TranscriptomeSAM   --readFilesCommand zcat \\\n",
    "     --readFilesIn SRX7253170_filt_1.fq.gz \\\n",
    "                   SRX7253170_filt_2.fq.gz\n",
    "\n",
    "#Ctlr3\n",
    "STAR --runMode alignReads      \\\n",
    "     --genomeDir /input_dir/corona_analysis/annotations/human/STAR_ix \\\n",
    "     --outSAMtype BAM SortedByCoordinate --runThreadN 14 \\\n",
    "     --outFileNamePrefix /input_dir/corona_analysis/alignment_out/BALF_control_quant/BALF_control_s3_\\\n",
    "     --outTmpDir /input_dir/corona_analysis/temp/star2 \\\n",
    "     --outReadsUnmapped Fastx --limitBAMsortRAM 27000000000 \\\n",
    "     --outWigType wiggle --quantMode TranscriptomeSAM   --readFilesCommand zcat \\\n",
    "     --readFilesIn SRX7253168_filt_1.fq.gz \\\n",
    "                   SRX7253168_filt_2.fq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd /input_dir/corona_analysis/alignment_out/BALF_control_quant/\n",
    "#Quantify expression in control BALF samples 1-3\n",
    "\n",
    "#Ctlr1\n",
    "rsem-calculate-expression --paired-end \\\n",
    "                       --alignments \\\n",
    "                       -p 14 --no-bam-output \\\n",
    "                       BALF_control_s1_Aligned.toTranscriptome.out.bam \\\n",
    "                      /input_dir/corona_analysis/annotations/human/RSEM_ix \\\n",
    "                       BALF_control_rsem_s1_\n",
    "#Ctlr2\n",
    "rsem-calculate-expression --paired-end \\\n",
    "                       --alignments \\\n",
    "                       -p 14 --no-bam-output \\\n",
    "                       BALF_control_s2_Aligned.toTranscriptome.out.bam \\\n",
    "                      /input_dir/corona_analysis/annotations/human/RSEM_ix \\\n",
    "                       BALF_control_rsem_s2_\n",
    "#Ctlr3\n",
    "rsem-calculate-expression --paired-end \\\n",
    "                       --alignments \\\n",
    "                       -p 14 --no-bam-output \\\n",
    "                       BALF_control_s3_Aligned.toTranscriptome.out.bam \\\n",
    "                      /input_dir/corona_analysis/annotations/human/RSEM_ix \\\n",
    "                       BALF_control_rsem_s3_\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get scRNA-seq (and scTCR-seq) from BALF CoVid patients\n",
    "\n",
    "## Paper: [Single-cell landscape of bronchoalveolar immune cells in patients with COVID-19](https://www.nature.com/articles/s41591-020-0901-9#additional-information)\n",
    "\n",
    "### Bioproject: PRJNA608742\n",
    "\n",
    "Input metadata for raw data stored in PRJNA608742_metadata.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-7-02ebf417dc20>:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  PRJNA608742_meta_in = pd.read_csv(\"PRJNA608742_metadata.txt\", header=1, delimiter=r'\\t+')\n"
     ]
    }
   ],
   "source": [
    "PRJNA608742_meta_in = pd.read_csv(\"PRJNA608742_metadata.txt\", header=1, delimiter=r'\\t+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SRR11537951</th>\n",
       "      <th>RNA-Seq</th>\n",
       "      <th>Total cell</th>\n",
       "      <th>SRX8108997</th>\n",
       "      <th>GSM4475053</th>\n",
       "      <th>severe COVID-19 patient</th>\n",
       "      <th>GSM4475053.1</th>\n",
       "      <th>SRP250732</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR11181954</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7802546</td>\n",
       "      <td>GSM4339769</td>\n",
       "      <td>mild</td>\n",
       "      <td>GSM4339769</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR11181955</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7802547</td>\n",
       "      <td>GSM4339770</td>\n",
       "      <td>mild</td>\n",
       "      <td>GSM4339770</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR11181956</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7802548</td>\n",
       "      <td>GSM4339771</td>\n",
       "      <td>severe</td>\n",
       "      <td>GSM4339771</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR11181957</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7802549</td>\n",
       "      <td>GSM4339772</td>\n",
       "      <td>mild</td>\n",
       "      <td>GSM4339772</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR11181958</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7802550</td>\n",
       "      <td>GSM4339773</td>\n",
       "      <td>severe</td>\n",
       "      <td>GSM4339773</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRR11181959</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7802551</td>\n",
       "      <td>GSM4339774</td>\n",
       "      <td>severe</td>\n",
       "      <td>GSM4339774</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRR11537946</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>CD45+</td>\n",
       "      <td>SRX8108992</td>\n",
       "      <td>GSM4475048</td>\n",
       "      <td>healthy control</td>\n",
       "      <td>GSM4475048</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRR11537947</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>CD45+</td>\n",
       "      <td>SRX8108993</td>\n",
       "      <td>GSM4475049</td>\n",
       "      <td>healthy control</td>\n",
       "      <td>GSM4475049</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRR11537948</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>CD45+</td>\n",
       "      <td>SRX8108994</td>\n",
       "      <td>GSM4475050</td>\n",
       "      <td>healthy control</td>\n",
       "      <td>GSM4475050</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRR11537949</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX8108995</td>\n",
       "      <td>GSM4475051</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "      <td>GSM4475051</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRR11537950</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX8108996</td>\n",
       "      <td>GSM4475052</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "      <td>GSM4475052</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRR11537952</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX8108998</td>\n",
       "      <td>GSM4475054</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "      <td>GSM4475054</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SRR11537953</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX8108999</td>\n",
       "      <td>GSM4475055</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "      <td>GSM4475055</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SRR11537954</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX8109000</td>\n",
       "      <td>GSM4475056</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "      <td>GSM4475056</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRR11245351</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7856785</td>\n",
       "      <td>GSM4385990</td>\n",
       "      <td>mild</td>\n",
       "      <td>GSM4385990</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRR11245352</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7856786</td>\n",
       "      <td>GSM4385991</td>\n",
       "      <td>mild</td>\n",
       "      <td>GSM4385991</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SRR11245353</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7856787</td>\n",
       "      <td>GSM4385992</td>\n",
       "      <td>severe</td>\n",
       "      <td>GSM4385992</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SRR11245354</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7856788</td>\n",
       "      <td>GSM4385993</td>\n",
       "      <td>mild</td>\n",
       "      <td>GSM4385993</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SRR11245355</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7856789</td>\n",
       "      <td>GSM4385994</td>\n",
       "      <td>severe</td>\n",
       "      <td>GSM4385994</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SRR11245356</td>\n",
       "      <td>TCR-seq</td>\n",
       "      <td>Total cell</td>\n",
       "      <td>SRX7856790</td>\n",
       "      <td>GSM4385995</td>\n",
       "      <td>severe</td>\n",
       "      <td>GSM4385995</td>\n",
       "      <td>SRP250732</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    SRR11537951  RNA-Seq  Total cell  SRX8108997  GSM4475053  \\\n",
       "0   SRR11181954  RNA-Seq  Total cell  SRX7802546  GSM4339769   \n",
       "1   SRR11181955  RNA-Seq  Total cell  SRX7802547  GSM4339770   \n",
       "2   SRR11181956  RNA-Seq  Total cell  SRX7802548  GSM4339771   \n",
       "3   SRR11181957  RNA-Seq  Total cell  SRX7802549  GSM4339772   \n",
       "4   SRR11181958  RNA-Seq  Total cell  SRX7802550  GSM4339773   \n",
       "5   SRR11181959  RNA-Seq  Total cell  SRX7802551  GSM4339774   \n",
       "6   SRR11537946  RNA-Seq       CD45+  SRX8108992  GSM4475048   \n",
       "7   SRR11537947  RNA-Seq       CD45+  SRX8108993  GSM4475049   \n",
       "8   SRR11537948  RNA-Seq       CD45+  SRX8108994  GSM4475050   \n",
       "9   SRR11537949  RNA-Seq  Total cell  SRX8108995  GSM4475051   \n",
       "10  SRR11537950  RNA-Seq  Total cell  SRX8108996  GSM4475052   \n",
       "11  SRR11537952  TCR-seq  Total cell  SRX8108998  GSM4475054   \n",
       "12  SRR11537953  TCR-seq  Total cell  SRX8108999  GSM4475055   \n",
       "13  SRR11537954  TCR-seq  Total cell  SRX8109000  GSM4475056   \n",
       "14  SRR11245351  TCR-seq  Total cell  SRX7856785  GSM4385990   \n",
       "15  SRR11245352  TCR-seq  Total cell  SRX7856786  GSM4385991   \n",
       "16  SRR11245353  TCR-seq  Total cell  SRX7856787  GSM4385992   \n",
       "17  SRR11245354  TCR-seq  Total cell  SRX7856788  GSM4385993   \n",
       "18  SRR11245355  TCR-seq  Total cell  SRX7856789  GSM4385994   \n",
       "19  SRR11245356  TCR-seq  Total cell  SRX7856790  GSM4385995   \n",
       "\n",
       "    severe COVID-19 patient GSM4475053.1  SRP250732  \n",
       "0                      mild   GSM4339769  SRP250732  \n",
       "1                      mild   GSM4339770  SRP250732  \n",
       "2                    severe   GSM4339771  SRP250732  \n",
       "3                      mild   GSM4339772  SRP250732  \n",
       "4                    severe   GSM4339773  SRP250732  \n",
       "5                    severe   GSM4339774  SRP250732  \n",
       "6           healthy control   GSM4475048  SRP250732  \n",
       "7           healthy control   GSM4475049  SRP250732  \n",
       "8           healthy control   GSM4475050  SRP250732  \n",
       "9   severe COVID-19 patient   GSM4475051  SRP250732  \n",
       "10  severe COVID-19 patient   GSM4475052  SRP250732  \n",
       "11  severe COVID-19 patient   GSM4475054  SRP250732  \n",
       "12  severe COVID-19 patient   GSM4475055  SRP250732  \n",
       "13  severe COVID-19 patient   GSM4475056  SRP250732  \n",
       "14                     mild   GSM4385990  SRP250732  \n",
       "15                     mild   GSM4385991  SRP250732  \n",
       "16                   severe   GSM4385992  SRP250732  \n",
       "17                     mild   GSM4385993  SRP250732  \n",
       "18                   severe   GSM4385994  SRP250732  \n",
       "19                   severe   GSM4385995  SRP250732  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PRJNA608742_meta_in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2020-07-25T23:54:16 prefetch.2.10.3: 1) 'SRR11181956' (95GB) is larger than maximum allowed: skipped \n",
      "\n",
      "Download of some files was skipped because they are too large\n",
      "You can change size download limit by setting\n",
      "--min-size and --max-size command line arguments\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2020-07-25T23:54:16 prefetch.2.10.3 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-07-25T23:54:16 prefetch.2.10.3 warn: Maximum file size download limit is 20GB \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Prefetch BALF covid\n",
    "#Do one severe patient\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/lung_scRNA_BALF/SRR11181956.sra \\\n",
    "    --max_size 100000000 SRR11181956\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#BALF covid\n",
    "parallel-fastq-dump -t 12 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/lung_scRNA_BALF/SRR11181956.sra \\\n",
    "    --dumpbase --clip --skip-technical --readids  \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/lung_scRNA_BALF\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "### optional (tbd) Fastp based read filtering\n",
    "fastp -p 8 -i /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR11181956_pass_1.fastq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/lung_scRNA_BALF/SRR11181956_pass_2.fastq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/lung_scRNA_BALF/SRR11181956_1_filt.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/lung_scRNA_BALF/SRR11181956_2_filt.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/lung_scRNA_BALF/SRR11181956_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/lung_scRNA_BALF/SRR11181956_fastp.json \\\n",
    "      --umi --umi_loc read1 --umi_len 10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir /input_dir/corona_analysis/alignment_out/lung_scRNA_BALF/\n",
    "\n",
    "#Align reads against human transcriptome for BALF CoVid \n",
    "cd /input_dir/corona_analysis/alignment_out/lung_scRNA_BALF\n",
    "\n",
    "#Severe covid sample1\n",
    "STAR --runMode alignReads      \\\n",
    "     --genomeDir /input_dir/corona_analysis/annotations/human/STAR_ix \\\n",
    "     --outSAMtype BAM SortedByCoordinate --runThreadN 14 \\\n",
    "     --outFileNamePrefix /input_dir/corona_analysis/alignment_out/lung_scRNA_BALF/lung_scRNA_covid_SRR11181956_ \\\n",
    "     --outTmpDir /input_dir/corona_analysis/temp/star2 \\\n",
    "     --outReadsUnmapped Fastx --limitBAMsortRAM 27000000000 \\\n",
    "     --outWigType wiggle --quantMode TranscriptomeSAM  \\\n",
    "#     --soloType CB_UMI_Complex --soloCBwhitelist None --soloCBposition  0_0_0_6 0_22_0_27 0_43_0_48 --soloUMIposition  0_49_0_54 \\\n",
    "     --soloFeatures Gene --soloOutFileNames lung_scRNA_covid_SRR11181956_solo_ \\\n",
    "     --readFilesIn SRR11181956_filt_1.fq.gz \\\n",
    "                   SRR11181956_filt_2.fq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd /input_dir/corona_analysis/alignment_out/lung_scRNA_BALF\n",
    "\n",
    "#Attempt doing a severe scRNA-seq CoVid sample\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR11181956_pass_1.fastq \\\n",
    "              -2 SRR11181956_pass_2.fastq \\\n",
    "              --chromium \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon/salmon_hg38_index \\\n",
    "              -p 3 \\\n",
    "              --mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/lung_scRNA_covid_/lung_scRNA_covid_SRR11181956_salmon \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get transcriptome for infected BALFs\n",
    "\n",
    "Transcriptome data comes in the form of RNA-seq from paper [Xiong et al 2020](https://www.tandfonline.com/doi/full/10.1080/22221751.2020.1747363). 2 patients Brancheolar Lavage Fluid (BALF) had RNA-seq libraries generated and sequenced on MiSeq\n",
    "CRR119897 - patient2_rep2\n",
    "CRR119896 - patient2_rep1\n",
    "CRR119895 - patient1_rep2\n",
    "CRR119894 - patient1_rep1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir /input_dir/corona_analysis/input_reads/BALF_rna\n",
    "\n",
    "#Get fq.gz from database for \n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119897/CRR119897_f1.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_f1.fq.gz\n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119897/CRR119897_r2.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_r2.fq.gz\n",
    "    \n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119896/CRR119896_f1.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_f1.fq.gz\n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119896/CRR119896_r2.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_r2.fq.gz\n",
    "    \n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119895/CRR119895_f1.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_f1.fq.gz\n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119895/CRR119895_r2.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_r2.fq.gz\n",
    "    \n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119894/CRR119894_f1.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_f1.fq.gz\n",
    "curl -s -L ftp://download.big.ac.cn/gsa3/CRA002390/CRR119894/CRR119894_r2.fq.gz  > \\\n",
    "    /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_r2.fq.gz\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter reads and QC runs using Fastp\n",
    "\n",
    "fastp -p 8 -i /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_f1.fq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_r2.fq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_filt_1.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_filt_2.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/BALF_rna/CRR119897_fastp.json \n",
    "\n",
    "fastp -p 8 -i /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_f1.fq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_r2.fq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_filt_1.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_filt_2.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/BALF_rna/CRR119896_fastp.json \n",
    "\n",
    "fastp -p 8 -i /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_f1.fq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_r2.fq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_filt_1.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_filt_2.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/BALF_rna/CRR119895_fastp.json \n",
    "\n",
    "fastp -p 8 -i /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_f1.fq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_r2.fq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_filt_1.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_filt_2.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/BALF_rna/CRR119894_fastp.json \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Align reads to human transcriptome using STAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Align coronavirus BALF reads to human transcriptome\n",
    "cd /input_dir/corona_analysis/input_reads/BALF_rna\n",
    "#Patient 1\n",
    "STAR --runMode alignReads      \\\n",
    "     --genomeDir /input_dir/corona_analysis/annotations/human/STAR_ix \\\n",
    "     --outSAMtype BAM SortedByCoordinate --runThreadN 14 \\\n",
    "     --outFileNamePrefix /input_dir/corona_analysis/alignment_out/BALF_corona_s1_\\\n",
    "     --outTmpDir /input_dir/corona_analysis/temp/star2 \\\n",
    "     --outReadsUnmapped Fastx --limitBAMsortRAM 27000000000 \\\n",
    "     --outWigType wiggle --quantMode TranscriptomeSAM   --readFilesCommand zcat \\\n",
    "     --readFilesIn CRR119894_filt_1.fq.gz,CRR119895_filt_1.fq.gz \\\n",
    "                   CRR119894_filt_2.fq.gz,CRR119895_filt_2.fq.gz\n",
    "#Patient 2\n",
    "STAR --runMode alignReads      \\\n",
    "     --genomeDir /input_dir/corona_analysis/annotations/human/STAR_ix \\\n",
    "     --outSAMtype BAM SortedByCoordinate --runThreadN 14 \\\n",
    "     --outFileNamePrefix /input_dir/corona_analysis/alignment_out/BALF_corona_s2_\\\n",
    "     --outTmpDir /input_dir/corona_analysis/temp/star2 \\\n",
    "     --outReadsUnmapped Fastx --limitBAMsortRAM 27000000000 \\\n",
    "     --outWigType wiggle --quantMode TranscriptomeSAM   --readFilesCommand zcat \\\n",
    "     --readFilesIn CRR119896_filt_1.fq.gz,CRR119897_filt_1.fq.gz \\\n",
    "                   CRR119896_filt_2.fq.gz,CRR119897_filt_2.fq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cd /input_dir/corona_analysis/alignment_out/\n",
    "#Quantify infected BALF transcriptome\n",
    "rsem-calculate-expression --paired-end \\\n",
    "                       --alignments \\\n",
    "                       -p 14 --no-bam-output \\\n",
    "                       BALF_corona_s1_Aligned.toTranscriptome.out.bam \\\n",
    "                      /input_dir/corona_analysis/annotations/human/RSEM_ix \\\n",
    "                       BALF_infected_s1_\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Quantify infected BALF transcriptome\n",
    "rsem-calculate-expression --paired-end \\\n",
    "                       --alignments \\\n",
    "                       -p 14 --no-bam-output \\\n",
    "                       BALF_corona_s2_Aligned.toTranscriptome.out.bam \\\n",
    "                      /input_dir/corona_analysis/annotations/human/RSEM_ix \\\n",
    "                       BALF_infected_s2_\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential expression between vero normal and infected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /input_dir/corona_analysis/alignment_out/\n",
    "rsem-generate-data-matrix BALF_control_quant/BALF_control_rsem_s1_.genes.results \\\n",
    "                          BALF_control_quant/BALF_control_rsem_s2_.genes.results \\\n",
    "                          BALF_control_quant/BALF_control_rsem_s3_.genes.results \\\n",
    "                          BALF_infected_quant/BALF_infected_s1_.genes.results \\\n",
    "                          BALF_infected_quant/BALF_infected_s2_.genes.results \\\n",
    "                            > BALF_rsem_gene_counts.txt\n",
    "\n",
    "rsem-generate-data-matrix BALF_control_quant/BALF_control_rsem_s1_.isoforms.results \\\n",
    "                          BALF_control_quant/BALF_control_rsem_s2_.isoforms.results \\\n",
    "                          BALF_control_quant/BALF_control_rsem_s3_.isoforms.results \\\n",
    "                          BALF_infected_quant/BALF_infected_s1_.isoforms.results \\\n",
    "                          BALF_infected_quant/BALF_infected_s2_.isoforms.results \\\n",
    "                            > BALF_rsem_isoform_counts.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /input_dir/corona_analysis/diffExp_out/\n",
    "rsem-run-ebseq BALF_rsem_gene_counts.txt 3,2 BALF_corona_geneDiff_rsem.txt"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
