{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PBMCs from healthy and CoVid\n",
    "\n",
    "[Immunophenotyping of COVID-19 and influenza highlights the role of type I interferons in development of severe COVID-19](https://immunology.sciencemag.org/content/5/49/eabd1554)\n",
    "\n",
    "[GEO dataset GSE149689](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149689)\n",
    "\n",
    "[SRA dataset PRJNA629752](https://www.ncbi.nlm.nih.gov/Traces/study/?query_key=2&WebEnv=MCID_5fef99ec1b3f1452c11c8a61&o=subject_group_sam_s%3Aa)\n",
    "\n",
    "Used Chromium V3 chemistry from 10x Genomics for scRNA-seq w/ readlayout:\n",
    "r1(UMI+CB)=28\n",
    "r2=91"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "PRJNA629752_meta_in = pd.read_csv(\"PRJNA629752_SRA_datatable.txt\", header=0, delimiter=r',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Run', 'AGE', 'Assay Type', 'AvgSpotLen', 'Bases', 'BioProject',\n",
       "       'BioSample', 'Bytes', 'Cell_type', 'Center Name', 'Consent',\n",
       "       'DATASTORE filetype', 'DATASTORE provider', 'DATASTORE region',\n",
       "       'Experiment', 'gender', 'GEO_Accession (exp)', 'Instrument',\n",
       "       'LibraryLayout', 'LibrarySelection', 'LibrarySource', 'Organism',\n",
       "       'Platform', 'ReleaseDate', 'Sample Name', 'source_name', 'SRA Study',\n",
       "       'subject_group', 'subject_status'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PRJNA629752_meta_in.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Run</th>\n",
       "      <th>subject_group</th>\n",
       "      <th>subject_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRX8241106</td>\n",
       "      <td>SRR11680207</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRX8241107</td>\n",
       "      <td>SRR11680208</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>mild COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRX8241108</td>\n",
       "      <td>SRR11680209</td>\n",
       "      <td>Influenza patient</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRX8241109</td>\n",
       "      <td>SRR11680210</td>\n",
       "      <td>Influenza patient</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRX8241110</td>\n",
       "      <td>SRR11680211</td>\n",
       "      <td>healthy control</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRX8241111</td>\n",
       "      <td>SRR11680212</td>\n",
       "      <td>Influenza patient</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRX8241112</td>\n",
       "      <td>SRR11680213</td>\n",
       "      <td>Influenza patient</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRX8241113</td>\n",
       "      <td>SRR11680214</td>\n",
       "      <td>Influenza patient</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRX8241114</td>\n",
       "      <td>SRR11680215</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRX8241115</td>\n",
       "      <td>SRR11680216</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRX8241116</td>\n",
       "      <td>SRR11680217</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>mild COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRX8241117</td>\n",
       "      <td>SRR11680218</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>mild COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SRX8241118</td>\n",
       "      <td>SRR11680219</td>\n",
       "      <td>healthy control</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SRX8241119</td>\n",
       "      <td>SRR11680220</td>\n",
       "      <td>healthy control</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRX8241120</td>\n",
       "      <td>SRR11680221</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRX8241121</td>\n",
       "      <td>SRR11680222</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SRX8241122</td>\n",
       "      <td>SRR11680223</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>severe COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SRX8241123</td>\n",
       "      <td>SRR11680224</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>mild COVID-19 patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SRX8241124</td>\n",
       "      <td>SRR11680225</td>\n",
       "      <td>healthy control</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SRX8241125</td>\n",
       "      <td>SRR11680226</td>\n",
       "      <td>COVID-19 patient</td>\n",
       "      <td>Asymptomatic case of COVID-19 patient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Experiment          Run      subject_group  \\\n",
       "0   SRX8241106  SRR11680207   COVID-19 patient   \n",
       "1   SRX8241107  SRR11680208   COVID-19 patient   \n",
       "2   SRX8241108  SRR11680209  Influenza patient   \n",
       "3   SRX8241109  SRR11680210  Influenza patient   \n",
       "4   SRX8241110  SRR11680211    healthy control   \n",
       "5   SRX8241111  SRR11680212  Influenza patient   \n",
       "6   SRX8241112  SRR11680213  Influenza patient   \n",
       "7   SRX8241113  SRR11680214  Influenza patient   \n",
       "8   SRX8241114  SRR11680215   COVID-19 patient   \n",
       "9   SRX8241115  SRR11680216   COVID-19 patient   \n",
       "10  SRX8241116  SRR11680217   COVID-19 patient   \n",
       "11  SRX8241117  SRR11680218   COVID-19 patient   \n",
       "12  SRX8241118  SRR11680219    healthy control   \n",
       "13  SRX8241119  SRR11680220    healthy control   \n",
       "14  SRX8241120  SRR11680221   COVID-19 patient   \n",
       "15  SRX8241121  SRR11680222   COVID-19 patient   \n",
       "16  SRX8241122  SRR11680223   COVID-19 patient   \n",
       "17  SRX8241123  SRR11680224   COVID-19 patient   \n",
       "18  SRX8241124  SRR11680225    healthy control   \n",
       "19  SRX8241125  SRR11680226   COVID-19 patient   \n",
       "\n",
       "                           subject_status  \n",
       "0                 severe COVID-19 patient  \n",
       "1                   mild COVID-19 patient  \n",
       "2                                     NaN  \n",
       "3                                     NaN  \n",
       "4                                     NaN  \n",
       "5                                     NaN  \n",
       "6                                     NaN  \n",
       "7                                     NaN  \n",
       "8                 severe COVID-19 patient  \n",
       "9                 severe COVID-19 patient  \n",
       "10                  mild COVID-19 patient  \n",
       "11                  mild COVID-19 patient  \n",
       "12                                    NaN  \n",
       "13                                    NaN  \n",
       "14                severe COVID-19 patient  \n",
       "15                severe COVID-19 patient  \n",
       "16                severe COVID-19 patient  \n",
       "17                  mild COVID-19 patient  \n",
       "18                                    NaN  \n",
       "19  Asymptomatic case of COVID-19 patient  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PRJNA629752_meta_in[[\"Experiment\",\"Run\",\"subject_group\", \"subject_status\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Start w/ severe CoVid patient SRR11680221"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Prefetch SRA file\n",
    "prefetch --progress -o /fast_dir/seq_data/raw_sra/SRR11680221.sra SRR11680221\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Dump SRA file into fastq.gz\n",
    "parallel-fastq-dump -t 8 --tmpdir /fast_dir/seq_data/raw_sra/temp \\\n",
    "    -s /fast_dir/seq_data/raw_sra/SRR11680221.sra \\\n",
    "    --dumpbase --clip --readids --gzip \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /fast_dir/seq_data/input_reads/pbmc/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /fast_dir/seq_data/input_reads/pbmc/\n",
    "#Filter reads and check quality, turn on UMI processing and max read length=0 to not toss first read\n",
    "fastp -p 12 -i SRR11680221_pass_1.fastq.gz \\\n",
    "      -I SRR11680221_pass_2.fastq.gz \\\n",
    "      -o SRR11680221_filt_1.fq.gz \\\n",
    "      -O SRR11680221_filt_2.fq.gz \\\n",
    "      -h SRR11680221_fastp.html \\\n",
    "      -j SRR11680221_fastp.json \n",
    "#      --umi --umi_loc read1 --umi_len 26 -l 0   #Finds and appends Barcode to read name, not what i want\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Quantify scRNA-seq reads\n",
    "cd /fast_dir/seq_data/input_reads/pbmc\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR11680221_pass_1.fastq.gz \\\n",
    "              -2 SRR11680221_pass_2.fastq.gz \\\n",
    "              --chromiumV3 \\\n",
    "              -i /data_dir/corona_analysis/annotations/human/salmon_ann/salmon_hg38_index \\\n",
    "              -p 10 --dumpMtx \\\n",
    "              --mrna /data_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /data_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -o /fast_dir/seq_data/alignment_out/pbmc/ \\\n",
    "              --tgMap /data_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
