{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import vpolo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make knee plot\n",
    "import pandas as pd\n",
    "cb_freq = pd.read_csv(\"/input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_jain_s1/alevin/raw_cb_frequency.txt\",\n",
    "                      sep=\"\\t\",index_col=0, header=None, names=[\"cell-barcode\",\"read_freq\"])\n",
    "                      \n",
    "                      \n",
    "import plotly.graph_objects as go\n",
    "import plotly.io as pio\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generKneePlot(cb_freq,\n",
    "                     save2disk=False,\n",
    "                     prefix_name=\"scRNA\"):\n",
    "     \n",
    "    #Toss those cell barcodes with less than 100 reads\n",
    "    cb_freq = cb_freq[cb_freq[\"read_freq\"] > 100]\n",
    "    ## 2D scatter for knee cell-barcode plot ##\n",
    "    xaxis=dict(\n",
    "        showline=False,\n",
    "        zeroline=False,\n",
    "        showgrid=False,\n",
    "        showticklabels=False,\n",
    "        showbackground=False,\n",
    "        ticks='',\n",
    "        title=\"Unique Cell-barcodes\")\n",
    "    yaxis=dict(\n",
    "        showline=False,\n",
    "        zeroline=False,\n",
    "        showgrid=False,\n",
    "        showticklabels=False,\n",
    "        showbackground=False,\n",
    "        type=\"log\",\n",
    "        ticks='',\n",
    "        title=\"log10 number of reads\")\n",
    "    layout = go.Layout(\n",
    "         scene = dict(\n",
    "                      xaxis=xaxis,\n",
    "                      yaxis=yaxis),\n",
    "        title=dict(text='Knee plot of Cell barcodes',\n",
    "            y=0.9,\n",
    "            x=0.5,\n",
    "            font=dict(size=30)),\n",
    "\n",
    "        legend = dict(x=.8,y=1),\n",
    "        showlegend=False,\n",
    "        width = 1200,\n",
    "        height = 800\n",
    "    )\n",
    "    cb_x = [int(x) for x in range(len(cb_freq.index))]\n",
    "    knee_trace = go.Scatter(\n",
    "        x=cb_x,\n",
    "        y=np.log10(cb_freq[\"read_freq\"]),\n",
    "        name='Knee plot of Cell barcodes',\n",
    "        mode=\"lines+markers\",\n",
    "        marker=go.scatter.Marker(color=\"red\",\n",
    "                                   opacity=1,\n",
    "                                   size=5,\n",
    "                                   line=dict(width=.4,\n",
    "                                             color='grey')),\n",
    "        text=[ str(cb_freq.index[x] + \":\" + str(cb_freq[\"read_freq\"][x])) for x in range(len(cb_freq[\"read_freq\"])-1) ],\n",
    "        hoverinfo=\"text\"\n",
    "    )\n",
    "\n",
    "    #Render output\n",
    "    data = [knee_trace]\n",
    "    fig = go.Figure(data=data, layout=layout)\n",
    "    fig.update_layout(template=\"plotly_white+xgridoff+ygridoff\")\n",
    "    if (save2disk):\n",
    "        pio.write_html(fig, file = prefix_name+'_knee_cb.html', auto_open=False)\n",
    "#        fig.update_layout(template=\"presentation+xgridoff+ygridoff\")\n",
    "#        fig.write_image(prefix_name+'_UMAP_.pdf')\n",
    "    \n",
    "    #Set up background for live viewing\n",
    "    fig.update_layout(template=\"plotly_dark+xgridoff+ygridoff\",\n",
    "                      paper_bgcolor='rgba(0,0,0,0)',\n",
    "                      plot_bgcolor='rgba(0,0,0,0)',\n",
    "                      width = 600, margin=dict(l=0,r=0,b=0,t=20))\n",
    "    return fig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "knee_out = generKneePlot(cb_freq,save2disk=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "kidney_s1_control = parser.read_quants_bin(\"/input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out/\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get version from paper\n",
    "!wget https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4008nnn/GSM4008621/suppl/GSM4008621_Adult-Kidney4-1_dge.txt.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#Read in from the paper text out\n",
    "kidney_s1_control_paper = pd.read_csv(\"/input_dir/corona_analysis/alignment_out/kidney_scRNA_control/paper_out/GSM4008621_Adult-Kidney4-1_dge.txt\",\n",
    "                                     sep=\"\\t\", index_col=0, header=0)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import plotly.express as px\n",
    "\n",
    "df = kidney_s1_control[\"ENSG00000198938.2\"]\n",
    "fig = px.violin(df, y=\"ENSG00000198938.2\")\n",
    "fig.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#See top most expressed genes on average\n",
    "(kidney_s1_control.mean(axis=0)).sort_values()[-5:]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "#Simple function to return rgb rainbow values for given categories\n",
    "def rainbow_cols(cats):\n",
    "    uniq_cats = np.unique(cats)\n",
    "    mult_rain = (int(len(uniq_cats)/9))+1\n",
    "    col_map = px.colors.sequential.Rainbow * mult_rain\n",
    "    chrom_rainbow = {uniq_cats[i]: col_map[i] for i in range(0,len(uniq_cats))}\n",
    "    mapped_cols = [chrom_rainbow[x] for x in cats]\n",
    "    return(mapped_cols)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import plotly.graph_objects as go\n",
    "import plotly.io as pio\n",
    "\n",
    "\n",
    "def generateUMAPplot(scRNA_inframe,\n",
    "                     save2disk=False,\n",
    "                     prefix_name=\"\",\n",
    "                    hgl_gene=\"none\"):\n",
    "    # Add labels for clusters and top 5 genes\n",
    "    # Add coloring by input gene\n",
    "    ## 3D Scatter for UMAP ##\n",
    "    axis=dict(\n",
    "        showline=False,\n",
    "        zeroline=False,\n",
    "        showgrid=False,\n",
    "        showticklabels=False,\n",
    "        showbackground=False,\n",
    "        ticks='',\n",
    "        title=\"UMAP dim\")\n",
    "    layout = go.Layout(\n",
    "         scene = dict(\n",
    "                      xaxis=axis,\n",
    "                      yaxis=axis,\n",
    "                      zaxis=axis),\n",
    "        title=dict(text='UMAP of gene <br> clustered w/ HDBSCAN',\n",
    "            y=0.9,\n",
    "            x=0.5,\n",
    "            font=dict(size=30)),\n",
    "\n",
    "        bargap=0.2,\n",
    "        bargroupgap=0.1,\n",
    "        legend = dict(x=.8,y=1),\n",
    "        showlegend=False,\n",
    "        width = 1400,\n",
    "        height = 900\n",
    "    )\n",
    "#    labels_text = TAD_labels(tad_features)\n",
    "    hdb_labels = scRNA_inframe[\"UMAP_hdbscan_labels\"]\n",
    "    hdb_labels_cols = rainbow_cols(hdb_labels)\n",
    "    \n",
    "    UMAP_trace = go.Scatter3d(\n",
    "        x=scRNA_inframe[\"UMAP_1\"],\n",
    "        y=scRNA_inframe[\"UMAP_2\"],\n",
    "        z=scRNA_inframe[\"UMAP_3\"],\n",
    "        name='UMAP of scRNA-seq genes',\n",
    "        mode=\"markers\",\n",
    "        marker=go.scatter3d.Marker(color=hdb_labels_cols,\n",
    "                                   opacity=1,\n",
    "                                   size=5,\n",
    "                                   line=dict(width=.4,\n",
    "                                             color='grey')),\n",
    "#        text=labels_text,\n",
    "        hoverinfo=\"text\"\n",
    "    )\n",
    "\n",
    "    #Render output\n",
    "    data = [UMAP_trace]\n",
    "    fig = go.Figure(data=data, layout=layout)\n",
    "    fig.update_layout(template=\"plotly_white+xgridoff+ygridoff\")\n",
    "    if (save2disk):\n",
    "        pio.write_html(fig, file = prefix_name+'_UMAP_genes.html', auto_open=False)\n",
    "#        fig.update_layout(template=\"presentation+xgridoff+ygridoff\")\n",
    "#        fig.write_image(prefix_name+'_UMAP_.pdf')\n",
    "    \n",
    "    #Set up background for live viewing\n",
    "    fig.update_layout(template=\"plotly_dark+xgridoff+ygridoff\",\n",
    "                      paper_bgcolor='rgba(0,0,0,0)',\n",
    "                      plot_bgcolor='rgba(0,0,0,0)',\n",
    "                      width = 600, margin=dict(l=0,r=0,b=0,t=20))\n",
    "    return fig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generatePercVar(var_out,\n",
    "                    save2disk=False,\n",
    "                    prefix_name=\"\"):\n",
    "    perc_explained = var_out[\"Percentage_variance\"]\n",
    "    ## Cumulative percent variance explained by PCA line plot ##\n",
    "    layout = go.Layout(\n",
    "        title=dict(text='Explained variance by <br> different principal components',\n",
    "            y=0.9,\n",
    "            x=0.5,\n",
    "            font=dict(size=30)),\n",
    "        xaxis=dict(\n",
    "            title='Explained Variance'\n",
    "        ),\n",
    "        yaxis=dict(\n",
    "            title='Explained variance in percent'\n",
    "        ),\n",
    "        bargap=0.2,\n",
    "        bargroupgap=0.1,\n",
    "        legend = dict(x=.8,y=1)\n",
    "    )\n",
    "\n",
    "    pca_trace = dict(\n",
    "        type='scatter',\n",
    "        x=['PC %s' %i for i in range(1,len(var_out))], \n",
    "        y=perc_explained,\n",
    "        name='Cumulative'\n",
    "    )\n",
    "\n",
    "    data = [pca_trace]\n",
    "    fig = dict(data=data, layout=layout)\n",
    "    fig = go.Figure(data=data, layout=layout)\n",
    "    fig.update_layout(template=\"plotly_white+ygridoff\")\n",
    "    if (save2disk):\n",
    "        pio.write_html(fig, file = prefix_name+'_PCA_variance.html', auto_open=False)\n",
    "        #fig.update_layout(template=\"presentation+ygridoff\")\n",
    "        #fig.write_image(prefix_name+'_PCA_variance.pdf')\n",
    "\n",
    "    #Set up background for live viewing\n",
    "    fig.update_layout(template=\"plotly_dark+xgridoff+ygridoff\",\n",
    "                      paper_bgcolor='rgba(0,0,0,0)',\n",
    "                      plot_bgcolor='rgba(0,0,0,0)',\n",
    "                      width = 600)\n",
    "    return fig\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generatePropVar(var_out,\n",
    "                    save2disk=False,\n",
    "                    prefix_name=\"\"):\n",
    "\n",
    "    prop_var = var_out[\"Proportion_variance\"]\n",
    "\n",
    "    layout = go.Layout(\n",
    "        title=dict(text='Scree Plot of scRNA-seq PCA',\n",
    "            y=0.95,\n",
    "            x=0.5,\n",
    "            font=dict(size=30)),\n",
    "        xaxis=dict(\n",
    "            title='Principal Component'\n",
    "        ),\n",
    "        yaxis=dict(\n",
    "            title='Proportion of Variance Explained'\n",
    "        ),\n",
    "        bargap=0.2,\n",
    "        bargroupgap=0.1,\n",
    "        legend = dict(x=.8,y=1)\n",
    "    )\n",
    "\n",
    "    pca_trace = dict(\n",
    "        type='scatter',\n",
    "        x=['PC %s' %i for i in range(1,len(var_out))], \n",
    "        y=prop_var,\n",
    "        name='Per PC variance'\n",
    "    )\n",
    "\n",
    "    data = [pca_trace]\n",
    "    fig = dict(data=data, layout=layout)\n",
    "    fig = go.Figure(data=data, layout=layout)\n",
    "    fig.update_layout(template=\"plotly_white+ygridoff\")\n",
    "    if (save2disk):\n",
    "        pio.write_html(fig, file = prefix_name+'_PCA_scree.html', auto_open=False)\n",
    "        #fig.update_layout(template=\"presentation+ygridoff\")\n",
    "        #fig.write_image(prefix_name+'_PCA_scree.pdf')\n",
    "        \n",
    "    #Set up background for live viewing\n",
    "    fig.update_layout(template=\"plotly_dark+xgridoff+ygridoff\",\n",
    "                      paper_bgcolor='rgba(0,0,0,0)',\n",
    "                      plot_bgcolor='rgba(0,0,0,0)',\n",
    "                      width = 1200)\n",
    "    return fig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "import os\n",
    "import sys\n",
    "import numpy as np\n",
    "import umap\n",
    "from sklearn.decomposition import PCA as sklearnPCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "import hdbscan\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#in_mat_exp = kidney_s1_control_paper.transpose().to_numpy()\n",
    "in_mat_exp = kidney_s1_control.to_numpy()\n",
    "\n",
    "#Run PCA using scikit\n",
    "num_pcs = 200\n",
    "X_std = StandardScaler().fit_transform(in_mat_exp)\n",
    "sklearn_pca = sklearnPCA(n_components=num_pcs)\n",
    "Y_sklearn = sklearn_pca.fit_transform(X_std)\n",
    "prop_var = sklearn_pca.explained_variance_ratio_\n",
    "perc_explained = np.cumsum(prop_var)\n",
    "var_out = pd.DataFrame({\"Percentage_variance\":perc_explained, \"Proportion_variance\":prop_var})\n",
    "#Save PCs to txt file for later viewing\n",
    "#PCs_out = pd.DataFrame(sklearn_pca.components_[0:num_pcs,:], columns=kidney_s1_control.columns,index = [\"PC-\" + str(i) for i in range(1,num_pcs+1)])\n",
    "#PCs_out.to_csv(prefix_name+\"_PCs.txt\", sep='\\t', header=True, index=False)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "generatePropVar(var_out, save2disk=True, prefix_name=\"/data/Euplotid/Scree_controlKidney\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "generatePercVar(var_out, save2disk=True, prefix_name=\"/data/Euplotid/CumVar_controlKidney\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make UMAP embedding of PCs and cluster using HDBSCAN\n",
    "umap_embedding = umap.UMAP(\n",
    "    n_neighbors=3,\n",
    "    min_dist=0.05,\n",
    "    n_components=3,\n",
    "    random_state=42,\n",
    "    metric='minkowski'\n",
    ").fit_transform(Y_sklearn[:,0:num_pcs])\n",
    "\n",
    "#Get clusters using HDBSCAN\n",
    "hdb_labels = hdbscan.HDBSCAN(\n",
    "    min_samples=5,\n",
    "    min_cluster_size=10,\n",
    ").fit_predict(umap_embedding)\n",
    "\n",
    "#Save embeddings along w/ feature matrix\n",
    "PC_embeds = pd.DataFrame( {\"PC_1\":Y_sklearn[:,0],\n",
    "                           \"PC_2\":Y_sklearn[:,1],\n",
    "                           \"UMAP_1\":umap_embedding[:,0],\n",
    "                           \"UMAP_2\":umap_embedding[:,1],\n",
    "                           \"UMAP_3\":umap_embedding[:,2],\n",
    "                           \"UMAP_hdbscan_labels\":hdb_labels })\n",
    "#kidney_s1_control = pd.concat([ kidney_s1_control, PC_embeds], axis=1)\n",
    "#Output calculated files\n",
    "#var_out.to_csv(prefix_name+\"_PCA_exp.txt\", sep=\"\\t\", header=True, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Plot\n",
    "generateUMAPplot(PC_embeds, save2disk=True, prefix_name=\"/data/Euplotid/UMAP_controlKidney\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
