{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Code to get lung, heart, kidney control uninfected scRNA-seq \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Really good kidney dataset from:\n",
    "\n",
    "Single cell transcriptomics identifies focal segmental glomerulosclerosis remission endothelial biomarker\n",
    "\n",
    "https://insight.jci.org/articles/view/133267\n",
    "\n",
    "SRA: PRJNA591621\n",
    "\n",
    "No raw data provided\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Good scRNA-seq data from 3 donors \n",
    "\n",
    "[\"Single-cell RNA sequencing of human kidney\"](https://www.nature.com/articles/s41597-019-0351-8)\n",
    "\n",
    "SRR10377488-SRR10377490 10x Genomics v2 chemistry\n",
    "\n",
    "First read only in SRA file because of weird .bam encoding, can download original bam and flip to fastq using bamtofastq 10x genomics utility, then concatenate and maybe feed to salmon?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Try getting data from [Construction of a human cell landscape at single-cell level](https://www.nature.com/articles/s41586-020-2157-4)\n",
    "Large compendium of different tissues all in scRNA-seq, done using the Microwell-seq methodology worth incorporating it seems?\n",
    "\n",
    "SRA website: https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP214806\n",
    "\n",
    "## Kidney \n",
    "* SRR9887772 - Sample_1_1 - Adult kidney\n",
    "* SRR9887773 - Sample_1_2 - Adult kidney\n",
    "* SRR9887774 - Sample_2_1 - Adult kidney\n",
    "* SRR9887775 - Sample_3_1 - Adult kidney\n",
    "\n",
    "## Lung\n",
    "* SRR9887736 - Sample_3_1 - Adult lung\n",
    "* SRR9887737 - Sample_2_1 - Adult lung\n",
    "* SRR9887738 - Sample_3_3 - Adult lung\n",
    "* SRR9887739 - Sample_3_2 - Adult lung\n",
    "\n",
    "## Heart only 2 adult \n",
    "* SRR9843423 - Sample_1_1 - Adult heart\n",
    "* SRR9843424 - Sample_2_1 - Adult heart\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Third source\n",
    "\n",
    "[A single-nucleus RNA-sequencing pipeline to decipher the molecular anatomy and pathophysiology of human kidneys](https://www.nature.com/articles/s41467-019-10861-2)\n",
    "\n",
    "snDrop-seq Jain\n",
    "\n",
    "Run\tExperiment\n",
    "* SRR8114240 - SRX4940606\n",
    "* SRR8114241 - SRX4940607\n",
    "* SRR8114242 - SRX4940608\n",
    "* SRR8114243 - SRX4940608\n",
    "* SRR8114244 - SRX4940609\n",
    "* SRR8114247 - SRX4940610\n",
    "* SRR8114248 - SRX4940611\n",
    "* SRR8114249 - SRX4940611\n",
    "* SRR8114250 - SRX4940612\n",
    "* SRR8114251 - SRX4940612\n",
    "* SRR8114252 - SRX4940613\n",
    "* SRR8114253 - SRX4940613\n",
    "* SRR8114254 - SRX4940614\n",
    "* SRR8114255 - SRX4940614\n",
    "* SRR8114256 - SRX4940615\n",
    "* SRR8114257 - SRX4940615\n",
    "* SRR8114258 - SRX4940616\n",
    "* SRR8114259 - SRX4940616\n",
    "* SRR8114260 - SRX4940617\n",
    "* SRR8114261 - SRX4940617\n",
    "* SRR8114262 - SRX4940618\n",
    "* SRR8114263 - SRX4940618\n",
    "* SRR8114264 - SRX4940619\n",
    "* SRR8114265 - SRX4940619\n",
    "* SRR8114266 - SRX4940620\n",
    "* SRR8114267 - SRX4940620\n",
    "* SRR8114268 - SRX4940621\n",
    "* SRR8114269 - SRX4940621\n",
    "* SRR8114271 - SRX4940622\n",
    "* SRR8114272 - SRX4940623\n",
    "* SRR8114273 - SRX4940623\n",
    "* SRR8114276 - SRX4940625\n",
    "* SRR8114277 - SRX4940625\n",
    "* SRR8114278 - SRX4940626\n",
    "* SRR8114279 - SRX4940626\n",
    "* SRR8114280 - SRX4940627\n",
    "* SRR8114281 - SRX4940627\n",
    "* SRR8114282 - SRX4940628\n",
    "* SRR8114283 - SRX4940628\n",
    "* SRR8114284 - SRX4940629\n",
    "* SRR8114285 - SRX4940629\n",
    "* SRR8114286 - SRX4940630\n",
    "* SRR8114288 - SRX4940631\n",
    "* SRR8114289 - SRX4940631\n",
    "* SRR8114290 - SRX4940632\n",
    "* SRR8114291 - SRX4940632\n",
    "* SRR8114292 - SRX4940633\n",
    "* SRR8114294 - SRX4940634\n",
    "* SRR8114295 - SRX4940634\n",
    "* SRR8114296 - SRX4940635\n",
    "* SRR8114297 - SRX4940635\n",
    "* SRR8114298 - SRX4940636\n",
    "* SRR8114299 - SRX4940636\n",
    "* SRR8114300 - SRX4940637\n",
    "* SRR8114301 - SRX4940637\n",
    "* SRR8114302 - SRX4940638\n",
    "* SRR8114303 - SRX4940638\n",
    "* SRR8114305 - SRX4940639\n",
    "* SRR8114307 - SRX4940640\n",
    "* SRR8114308 - SRX4940641\n",
    "* SRR8114309 - SRX4940641\n",
    "* SRR8114311 - SRX4940642\n",
    "* SRR8114312 - SRX4940643\n",
    "* SRR8114314 - SRX4940644\n",
    "* SRR8114315 - SRX4940644\n",
    "* SRR8114316 - SRX4940645\n",
    "* SRR8114317 - SRX4940645\n",
    "* SRR8114318 - SRX4940646\n",
    "* SRR8114319 - SRX4940646\n",
    "* SRR8114320 - SRX4940647\n",
    "* SRR8114321 - SRX4940647\n",
    "* SRR8114322 - SRX4940648\n",
    "* SRR8114323 - SRX4940648\n",
    "* SRR8114324 - SRX4940649\n",
    "* SRR8114325 - SRX4940649\n",
    "* SRR8114326 - SRX4940650\n",
    "* SRR8114327 - SRX4940650\n",
    "* SRR8114328 - SRX4940651\n",
    "* SRR8114329 - SRX4940651\n",
    "* SRR8114331 - SRX4940652\n",
    "* SRR8114246 - SRX4940610\n",
    "* SRR8114270 - SRX4940622\n",
    "* SRR8114274 - SRX4940624\n",
    "* SRR8114275 - SRX4940624\n",
    "* SRR8114287 - SRX4940630\n",
    "* SRR8114304 - SRX4940639\n",
    "* SRR8114306 - SRX4940640\n",
    "* SRR8114310 - SRX4940642\n",
    "* SRR8114313 - SRX4940643\n",
    "* SRR8114330 - SRX4940652\n",
    "* SRR8114293 - SRX4940633\n",
    "* SRR8114245 - SRX4940609\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Prefetch normal kidney samples\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887772.sra SRR9887772\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887773.sra SRR9887773\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887774.sra SRR9887774\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887775.sra SRR9887775\n",
    "\n",
    "#Prefetch normal lung samples\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887736.sra SRR9887736\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887737.sra SRR9887737\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887738.sra SRR9887738\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887739.sra SRR9887739\n",
    "\n",
    "#Prefetch normal heart samples\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/heart_scRNA_control/SRR9843423.sra SRR9843423\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/heart_scRNA_control/SRR9843424.sra SRR9843424\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Prefetch normal kidney samples\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377488.sra SRR10377488\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377489.sra SRR10377489\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377490.sra SRR10377490\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Prefetch normal kidney samples https://www.nature.com/articles/s41467-019-10861-2\n",
    "prefetch -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR8114241.sra SRR8114241\n",
    "\t\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Kidney\n",
    "parallel-fastq-dump -t 12 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR8114241.sra \\\n",
    "    --dumpbase --clip --skip-technical --readids  \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#More kidney samples\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377488.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377489.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377490.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Flip SRA to fastq no gzip \n",
    "#Kidney\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887772.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887773.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887774.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887775.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n",
    "#Lung\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887736.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/lung_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887737.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/lung_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887738.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/lung_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/lung_scRNA_control/SRR9887739.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/lung_scRNA_control\n",
    "#Heart\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/heart_scRNA_control/SRR9843423.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/heart_scRNA_control\n",
    "parallel-fastq-dump -t 8 --tmpdir /input_dir/corona_analysis/temp/ \\\n",
    "    -s /input_dir/corona_analysis/input_reads/heart_scRNA_control/SRR9843424.sra \\\n",
    "    --dumpbase --clip --readids \\\n",
    "    --read-filter pass --split-files --origfmt \\\n",
    "    --outdir /input_dir/corona_analysis/input_reads/heart_scRNA_control\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Use UMI tools to pull out barcode whitelist\n",
    "umi_tools whitelist --bc-pattern CCCCCCXXXXXXXXXXXXXXXCCCCCCXXXXXXXXXXXXXXXCCCCCCNNNNNN \\\n",
    "    -I SRR9887762_pass_1.fastq.gz \\\n",
    "    -S test_barcode.txt --subset-reads 15000000\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Filter out adapters that don't have at least 9 of 10 Ts at the end\n",
    "\n",
    "~/.local/bin/cutadapt -j 10 -a TTTTTTTTT$ -m 21 -e 0.6 --discard-untrimmed \\\n",
    "    -o SRR8114241_filt_1.fastq \\\n",
    "    -p SRR8114241_filt_2.fastq \\\n",
    "     SRR8114241_pass_1.fastq SRR8114241_pass_2.fastq\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Get original uploaded .bam files and flip to fastq using cellranger software because SRA is dumb about scRNA-seq and scrubs the goddamn index read\n",
    "curl -s -L https://sra-pub-src-2.s3.amazonaws.com/SRR10377488/possorted_genome_bam1.bam.1 > \\\n",
    "    /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377488.bam\n",
    "curl -s -L https://sra-pub-src-2.s3.amazonaws.com/SRR10377489/possorted_genome_bam2.bam.1 > \\\n",
    "    /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377489.bam\n",
    "curl -s -L https://sra-pub-src-2.s3.amazonaws.com/SRR10377490/possorted_genome_bam3.bam.1 > \\\n",
    "    /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377490.bam\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /input_dir/corona_analysis/input_reads/kidney_scRNA_control/\n",
    "#Get app to flip bams from cellranger etc to fastq\n",
    "curl -s -L http://cf.10xgenomics.com/misc/bamtofastq-1.2.0 > bamtofastq\n",
    "chmod 755 bamtofastq\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Flip downloaded original cell ranger output to fq to rerun analysis\n",
    "./bamtofastq --nthreads 12 SRR10377488.bam /input_dir/corona_analysis/input_reads/kidney_scRNA_control/s1\n",
    "./bamtofastq --nthreads 12 SRR10377489.bam /input_dir/corona_analysis/input_reads/kidney_scRNA_control/s2\n",
    "./bamtofastq --nthreads 12 SRR10377490.bam /input_dir/corona_analysis/input_reads/kidney_scRNA_control/s3\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Concatenate read files together for index and read\n",
    "cd /input_dir/corona_analysis/input_reads/kidney_scRNA_control/s1/kidney1_result_0_1_H7K5MDMXX\n",
    "cat *I* > /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377488_I.fq.gz\n",
    "cat *R* > /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377488_R.fq.gz\n",
    "\n",
    "cd /input_dir/corona_analysis/input_reads/kidney_scRNA_control/s2/kidney2_result_0_1_HMKGYCCXY\n",
    "cat *I* > /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377489_I.fq.gz\n",
    "cat *R* > /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377489_R.fq.gz\n",
    "\n",
    "cd /input_dir/corona_analysis/input_reads/kidney_scRNA_control/s3/kidney3_result_0_1_HMKGYCCX\n",
    "cat *I* > /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377490_I.fq.gz\n",
    "cat *R* > /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR10377490_R.fq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Filter input RNA-seq reads from control scRNA kidney from donors\n",
    "#Ctrl1\n",
    "fastp -p 8 -i /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887762_pass_1.fastq.gz \\\n",
    "      -I /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887762_pass_2.fastq.gz \\\n",
    "      -o /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887762_1_filt.fq.gz \\\n",
    "      -O /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887762_2_filt.fq.gz \\\n",
    "      -h /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887762_fastp.html \\\n",
    "      -j /input_dir/corona_analysis/input_reads/kidney_scRNA_control/SRR9887762_fastp.json \\\n",
    "      --umi --umi_loc read1 --umi_len 54\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# K-Mer based workflow for scRNA-seq\n",
    "\n",
    "* Compile needed annotation files\n",
    "* Build needed references\n",
    "* Quantify against reference\n",
    "\n",
    "## Get and massage annotation files needed for Salmon human\n",
    "\n",
    "Using all Ensembl-99 as ground truth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd /input_dir/corona_analysis/annotations/human/\n",
    "\n",
    "#Get hard masked hg38 genome and filter for only canonical chromosomes\n",
    "curl -s -L ftp://ftp.ensembl.org/pub/release-99/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_rm.primary_assembly.fa.gz > \\\n",
    "    /input_dir/corona_analysis/annotations/human/Homo_sapiens.GRCh38.dna_rm.primary_assembly.fa.gz\n",
    "gunzip Homo_sapiens.GRCh38.dna_rm.primary_assembly.fa.gz\n",
    "#Toss everything except canonical chromosomes\n",
    "awk \"/^>/ {n++} n>25 {exit} {print}\" /input_dir/corona_analysis/annotations/human/Homo_sapiens.GRCh38.dna_rm.primary_assembly.fa > GRCh38_filt_dna_rm.fa\n",
    "#Get list of decoy sequences for salmon\n",
    "grep \"^>\" /input_dir/corona_analysis/annotations/human/GRCh38_filt_dna_rm.fa | cut -d \" \" -f 1 > decoys.txt\n",
    "sed -i.bak -e 's/>//g' decoys.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Get cDNA of transcripts for hg38\n",
    "wget ftp://ftp.ensembl.org/pub/release-99/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "gunzip Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "\n",
    "#Filter the transcripts for only those in STAR \n",
    "cut -f 1 /input_dir/corona_analysis/annotations/human/STAR_ix/transcriptInfo.tab > in_enst_names.txt\n",
    "awk '{ if ((NR>1)&&($0~/^>/)) { printf(\"\\n%s\", $0); } else if (NR==1) { printf(\"%s\", $0); } else { printf(\"\\t%s\", $0); } }' \\\n",
    "    Homo_sapiens.GRCh38.cdna.all.fa \\\n",
    "    | grep -Ff in_enst_names.txt - \\\n",
    "    | tr \"\\t\" \"\\n\" > Hg38_cdna_filt.fa\n",
    "\n",
    "# coding_genes_hgnc.txt = list of all HGNC protein coding symbols\n",
    "#Filter the transcripts to ONLY protein coding\n",
    "awk '{ if ((NR>1)&&($0~/^>/)) { printf(\"\\n%s\", $0); } else if (NR==1) { printf(\"%s\", $0); } else { printf(\"\\t%s\", $0); } }' \\\n",
    "    Hg38_cdna_filt.fa \\\n",
    "    | grep -Ff coding_genes_hgnc.txt - \\\n",
    "    | tr \"\\t\" \"\\n\" > Hg38_cdna_coding_filt.fa\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd /input_dir/corona_analysis/annotations/human/\n",
    "\n",
    "#Get ensembl gtf file\n",
    "curl -s -L ftp://ftp.ensembl.org/pub/release-99/gtf/homo_sapiens/Homo_sapiens.GRCh38.99.gtf.gz > \\\n",
    "    /input_dir/corona_analysis/annotations/human/Homo_sapiens.GRCh38.99.gtf.gz\n",
    "gunzip /input_dir/corona_analysis/annotations/human/Homo_sapiens.GRCh38.99.gtf.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get rRNA from \n",
    "\n",
    "https://www.genenames.org/data/genegroup/#!/group/848\n",
    "\n",
    "Flip to ENST using Biomart and feed into rRNA_ensembl.txt to get rRNA_ensembl_cdna.fa\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Make gentrome for Salmon\n",
    "mkdir /input_dir/corona_analysis/annotations/human/salmon_ann\n",
    "cd /input_dir/corona_analysis/annotations/human/salmon_ann\n",
    "cat /input_dir/corona_analysis/annotations/human/Hg38_cdna_coding_filt.fa \\\n",
    "    /input_dir/corona_analysis/annotations/human/rRNA_ensembl_cdna.fa \\\n",
    "    /input_dir/corona_analysis/annotations/human/GRCh38_filt_dna_rm.fa \\\n",
    "    > gentrome_hg38_filt.fa \n",
    "\n",
    "#Full cdna file: Homo_sapiens.GRCh38.cdna.all.fa\n",
    "#Filtered cdna file: Hg38_cdna_coding_filt.fa\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Reduce ram used w/ sparse\n",
    "salmon index --sparse -t gentrome_hg38_filt.fa -d ../decoys.txt -p 10 -i salmon_hg38_index\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtf_parse = \"/input_dir/corona_analysis/annotations/human/Homo_sapiens.GRCh38.99.gtf\"\n",
    "mt_tran_out = \"/input_dir/corona_analysis/annotations/human/gencode_mt.txt\"\n",
    "out_tran_map = \"/input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\"\n",
    "\n",
    "tran_gene_dict = dict()\n",
    "tran_name_dict = dict()\n",
    "\n",
    "with open(gtf_parse, \"r\") as gene_in:\n",
    "    with open(mt_tran_out,\"w\") as mito_out:\n",
    "        for line in gene_in:\n",
    "            if line.startswith(\"#\"):\n",
    "                continue\n",
    "            else:\n",
    "                arr = line.strip().split(\"\\t\")\n",
    "                if (arr[2] == \"transcript\"):\n",
    "                    tmp_arr = arr[-1].strip(';').strip().split(\";\")\n",
    "                    tmp_arr = tmp_arr[:-1]\n",
    "                    arr_tran = [feature.strip().split(\" \") for feature in tmp_arr]\n",
    "                    cur_tran_dict = {key: value for (key, value) in arr_tran}\n",
    "                    tran_gene_dict[cur_tran_dict[\"transcript_id\"]+\".\"+cur_tran_dict[\"transcript_version\"]] = cur_tran_dict[\"gene_id\"]+\".\"+cur_tran_dict[\"gene_version\"]\n",
    "                    tran_name_dict[cur_tran_dict[\"transcript_id\"]+\".\"+cur_tran_dict[\"transcript_version\"]] = cur_tran_dict[\"gene_name\"]\n",
    "\n",
    "                    if arr[0] == \"MT\":\n",
    "                        mito_out.write(cur_tran_dict[\"transcript_id\"].strip(\"\\\"\")+\".\"+cur_tran_dict[\"transcript_version\"].strip(\"\\\"\") + \"\\n\")\n",
    "                        \n",
    "                        \n",
    "\n",
    "with open(out_tran_map, \"w\") as kal_tran:\n",
    "    for tran in tran_gene_dict.keys():\n",
    "        out_line = tran.replace(\"\\\"\",\"\") + \"\\t\" + tran_gene_dict[tran].replace(\"\\\"\",\"\") + \"\\n\"\n",
    "        kal_tran.write(out_line)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Add rRNA to transcript2gene map\n",
    "grep \">\" /input_dir/corona_analysis/annotations/human/rRNA_ensembl_cdna.fa | \\\n",
    "    awk -F '|' '{print $4 \"\\t\" $2 }' >> /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Get transcript IDs of rRNA w/ version ID\n",
    "grep \">\" /input_dir/corona_analysis/annotations/human/rRNA_ensembl_cdna.fa | cut -f 1 -d '|' \\\n",
    "    | sed 's/>//' > \\\n",
    "    /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Map kidney scRNA reads using alevin\n",
    "cd /input_dir/corona_analysis/input_reads/kidney_scRNA_control\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#Attempt doing a sample from landscape\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR9887762_pass_1.fastq \\\n",
    "              -2 SRR9887762_pass_2.fastq \\\n",
    "              --dropseq \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon/salmon_hg38_index \\\n",
    "              -p 3 \\\n",
    "              --mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_s1 \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Attempt doing a sample from landscape\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR9887772_pass_1.fastq \\\n",
    "              -2 SRR9887772_pass_2.fastq \\\n",
    "              --dropseq \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon_ann/salmon_hg38_index \\\n",
    "              -p 10 \\\n",
    "              --mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_landscape_s1 \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n",
    "\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR9887773_pass_1.fastq \\\n",
    "              -2 SRR9887773_pass_2.fastq \\\n",
    "              --dropseq \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon_ann/salmon_hg38_index \\\n",
    "              -p 10 \\\n",
    "              --mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_landscape_s2 \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "##Figure out whitelisting issue\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR8114241_filt_1.fastq \\\n",
    "              -2 SRR8114241_filt_2.fastq \\\n",
    "              --dropseq \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon_ann/salmon_hg38_index \\\n",
    "              -p 10 --noQuant --dumpFeature \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_jain_s1 \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n",
    "\n",
    "\n",
    " \n",
    "#Run for snDrop-Seq jain one kidney using hard 2000 expectCells cutoff for CB\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR8114241_filt_1.fastq \\\n",
    "              -2 SRR8114241_filt_2.fastq \\\n",
    "              --dropseq --forceCells 58478 \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon_ann/salmon_hg38_index \\\n",
    "              -p 14 --dumpFeature \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_jain_s1 \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n",
    "\n",
    "\n",
    "# manual barcode info\n",
    "# --barcodeLength 12 --umiLength 8 --end 3\n",
    "\n",
    "# Mitochondrial and ribosomal RNA estimation of CB whitelisting\n",
    "--mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "--rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Map lung scRNA reads using alevin\n",
    "cd /input_dir/corona_analysis/input_reads/lung_scRNA_control\n",
    "\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR9887739_pass_1.fastq \\\n",
    "              -2 SRR9887739_pass_2.fastq \\\n",
    "              --dropseq \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon/salmon_hg38_index \\\n",
    "              -p 8 \\\n",
    "              --mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/lung_scRNA_control/salmon_out_1s \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    " %%bash \n",
    "    \n",
    "#Run for snDrop-Seq jain one kidney using hard 2000 expectCells cutoff for CB\n",
    "salmon alevin -l ISR \\\n",
    "              -1 SRR10377488_I.fq.gz \\\n",
    "              -2 SRR10377488_R.fq.gz \\\n",
    "              --chromium \\\n",
    "              --mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon_ann/salmon_hg38_index \\\n",
    "              -p 14 --dumpFeature \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_mo_s1 \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt\n",
    "\n",
    "\n",
    " %%bash \n",
    "    \n",
    "#Run for snDrop-Seq jain one kidney using hard 2000 expectCells cutoff for CB\n",
    "salmon alevin -l ISR \\\n",
    "              -1 t1.fq \\\n",
    "              -2 t2.fq \\\n",
    "              --chromium \\\n",
    "              --mrna /input_dir/corona_analysis/annotations/human/gencode_mt.txt \\\n",
    "              --rrna /input_dir/corona_analysis/annotations/human/rRNA_ensembl.txt \\\n",
    "              -i /input_dir/corona_analysis/annotations/human/salmon_ann/salmon_hg38_index \\\n",
    "              -p 3 --dumpFeature \\\n",
    "              -o /input_dir/corona_analysis/alignment_out/kidney_scRNA_control/salmon_out_mo_s1 \\\n",
    "              --tgMap /input_dir/corona_analysis/annotations/human/salmon_grch38_gencode_tran2gene.txt"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
